{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We get the subtypes for our patients using SNF algorithm. However, the algorithm can only splits the patients into groups with no information about which particular classes the groups belong to. Thus, here, we find the consensus between the true labels and our results. \n",
    "\n",
    "One route can be finding the differences between our grouping and the labels from TCGA. Our assumption here is that SNF tend to produce similar outputs as TCGA pipeline does.\n",
    "\n",
    "Another route is that we ignore the true label, continue analysis with our grouping to see if our grouping make any sense."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Evaluate SNF result comparing with TCGA subtype"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(347, 1)"
      ]
     },
     "execution_count": 230,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group = pd.read_csv('/Users/yuanqizhao/Desktop/SNFtool/Result/3clusters/group.txt',delimiter= ' ',index_col=0)\n",
    "group.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([3, 1, 2])"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group['x'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 232,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>GeneExp_Subtype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>TCGA-02-0001-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TCGA-02-0003-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TCGA-02-0007-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 GeneExp_Subtype\n",
       "TCGA-02-0001-01                1\n",
       "TCGA-02-0003-01                3\n",
       "TCGA-02-0007-01                3"
      ]
     },
     "execution_count": 232,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "TCGA_subtype = pd.read_csv('data/true_subtype.txt', header=0,index_col=0 , sep='\\t')\n",
    "TCGA_subtype[:3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 233,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>x</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>TCGA-02-0001-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TCGA-02-0003-01</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TCGA-02-0007-01</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 x\n",
       "TCGA-02-0001-01  3\n",
       "TCGA-02-0003-01  3\n",
       "TCGA-02-0007-01  1"
      ]
     },
     "execution_count": 233,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group.index = TCGA_subtype.index\n",
    "group[:3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "group.to_csv('/Users/yuanqizhao/Desktop/SNFtool/Result/SNF_subtype4.txt', header=False, index=True, sep='\\t', mode='a')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "TCGA_group = TCGA_subtype.groupby(['GeneExp_Subtype'])\n",
    "# TCGA_group.groups"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "SNF_group = group.groupby(['x'])\n",
    "# SNF_group.groups"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Use largest common elements to reassign labels"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[(0, 0.3858267716535433),\n",
       " (0, 0.2204724409448819),\n",
       " (3, 0.6078431372549019),\n",
       " (2, 0.5802469135802469)]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "best_group = []\n",
    "for i in TCGA_group.__iter__():\n",
    "    sub_tcga = set(list(i[1].index))\n",
    "    inter_len=[]\n",
    "    for j in SNF_group.__iter__():\n",
    "        sub_group = set(list(j[1].index))\n",
    "        inter = len(list(sub_tcga & sub_group))\n",
    "#         print(j[0],': ', inter)\n",
    "        inter_len.append(inter/len(sub_group))   ###percentage of match\n",
    "    max_per = np.argmax(inter_len)\n",
    "    best_group.append((max_per,inter_len[max_per]))\n",
    "best_group"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We changed to three clusters."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "https://www.nature.com/articles/nrclinonc.2017.122#:~:text=Glioblastoma%20(GBM)%20was%20previously%20classified,nonmalignant%20cells%20in%20the%20analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 3 Clusters clinical differences"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(347, 1)"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CDE_DxAge</th>\n",
       "      <th>CDE_alk_chemoradiation_standard</th>\n",
       "      <th>CDE_chemo_adjuvant_alk</th>\n",
       "      <th>CDE_chemo_adjuvant_tmz</th>\n",
       "      <th>CDE_chemo_alk</th>\n",
       "      <th>CDE_chemo_alk_days</th>\n",
       "      <th>CDE_chemo_alk_long</th>\n",
       "      <th>CDE_chemo_tmz</th>\n",
       "      <th>CDE_chemo_tmz_days</th>\n",
       "      <th>CDE_chemo_tmz_long</th>\n",
       "      <th>...</th>\n",
       "      <th>_GENOMIC_ID_TCGA_GBM_exp_u133a</th>\n",
       "      <th>_GENOMIC_ID_TCGA_GBM_exp_HiSeqV2_PANCAN</th>\n",
       "      <th>_GENOMIC_ID_TCGA_GBM_RPPA_RBN</th>\n",
       "      <th>_GENOMIC_ID_TCGA_GBM_gistic2thd</th>\n",
       "      <th>_GENOMIC_ID_TCGA_GBM_PDMRNAseqCNV</th>\n",
       "      <th>_GENOMIC_ID_TCGA_GBM_mutation</th>\n",
       "      <th>_GENOMIC_ID_TCGA_GBM_exp_HiSeqV2_percentile</th>\n",
       "      <th>_GENOMIC_ID_TCGA_GBM_exp_HiSeqV2_exon</th>\n",
       "      <th>_GENOMIC_ID_TCGA_GBM_gistic2</th>\n",
       "      <th>_GENOMIC_ID_TCGA_GBM_hMethyl450</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>sampleID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>TCGA-02-0001-01</td>\n",
       "      <td>44.30</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>03c14838-bbab-4487-80c3-c5fdf6ec83f9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-02-0001-01C-01D-0182-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-02-0001-01C-01D-0182-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TCGA-02-0002-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TCGA-02-0003-01</td>\n",
       "      <td>50.21</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>...</td>\n",
       "      <td>4298ad72-e8eb-4749-95da-cce602107724</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-02-0003-01A-21-1898-20</td>\n",
       "      <td>TCGA-02-0003-01A-01D-0182-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-02-0003-01A-01D-1490-08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-02-0003-01A-01D-0182-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 128 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 CDE_DxAge CDE_alk_chemoradiation_standard  \\\n",
       "sampleID                                                     \n",
       "TCGA-02-0001-01      44.30                           False   \n",
       "TCGA-02-0002-01        NaN                             NaN   \n",
       "TCGA-02-0003-01      50.21                           False   \n",
       "\n",
       "                CDE_chemo_adjuvant_alk CDE_chemo_adjuvant_tmz CDE_chemo_alk  \\\n",
       "sampleID                                                                      \n",
       "TCGA-02-0001-01                  False                  False         False   \n",
       "TCGA-02-0002-01                    NaN                    NaN           NaN   \n",
       "TCGA-02-0003-01                  False                  False         False   \n",
       "\n",
       "                 CDE_chemo_alk_days CDE_chemo_alk_long CDE_chemo_tmz  \\\n",
       "sampleID                                                               \n",
       "TCGA-02-0001-01                 0.0              False         False   \n",
       "TCGA-02-0002-01                 NaN                NaN           NaN   \n",
       "TCGA-02-0003-01                 0.0              False         False   \n",
       "\n",
       "                 CDE_chemo_tmz_days CDE_chemo_tmz_long  ...  \\\n",
       "sampleID                                                ...   \n",
       "TCGA-02-0001-01                 0.0              False  ...   \n",
       "TCGA-02-0002-01                 NaN                NaN  ...   \n",
       "TCGA-02-0003-01                 0.0              False  ...   \n",
       "\n",
       "                       _GENOMIC_ID_TCGA_GBM_exp_u133a  \\\n",
       "sampleID                                                \n",
       "TCGA-02-0001-01  03c14838-bbab-4487-80c3-c5fdf6ec83f9   \n",
       "TCGA-02-0002-01                                   NaN   \n",
       "TCGA-02-0003-01  4298ad72-e8eb-4749-95da-cce602107724   \n",
       "\n",
       "                _GENOMIC_ID_TCGA_GBM_exp_HiSeqV2_PANCAN  \\\n",
       "sampleID                                                  \n",
       "TCGA-02-0001-01                                     NaN   \n",
       "TCGA-02-0002-01                                     NaN   \n",
       "TCGA-02-0003-01                                     NaN   \n",
       "\n",
       "                _GENOMIC_ID_TCGA_GBM_RPPA_RBN _GENOMIC_ID_TCGA_GBM_gistic2thd  \\\n",
       "sampleID                                                                        \n",
       "TCGA-02-0001-01                           NaN    TCGA-02-0001-01C-01D-0182-01   \n",
       "TCGA-02-0002-01                           NaN                             NaN   \n",
       "TCGA-02-0003-01   TCGA-02-0003-01A-21-1898-20    TCGA-02-0003-01A-01D-0182-01   \n",
       "\n",
       "                _GENOMIC_ID_TCGA_GBM_PDMRNAseqCNV  \\\n",
       "sampleID                                            \n",
       "TCGA-02-0001-01                               NaN   \n",
       "TCGA-02-0002-01                               NaN   \n",
       "TCGA-02-0003-01                               NaN   \n",
       "\n",
       "                _GENOMIC_ID_TCGA_GBM_mutation  \\\n",
       "sampleID                                        \n",
       "TCGA-02-0001-01                           NaN   \n",
       "TCGA-02-0002-01                           NaN   \n",
       "TCGA-02-0003-01  TCGA-02-0003-01A-01D-1490-08   \n",
       "\n",
       "                _GENOMIC_ID_TCGA_GBM_exp_HiSeqV2_percentile  \\\n",
       "sampleID                                                      \n",
       "TCGA-02-0001-01                                         NaN   \n",
       "TCGA-02-0002-01                                         NaN   \n",
       "TCGA-02-0003-01                                         NaN   \n",
       "\n",
       "                _GENOMIC_ID_TCGA_GBM_exp_HiSeqV2_exon  \\\n",
       "sampleID                                                \n",
       "TCGA-02-0001-01                                   NaN   \n",
       "TCGA-02-0002-01                                   NaN   \n",
       "TCGA-02-0003-01                                   NaN   \n",
       "\n",
       "                 _GENOMIC_ID_TCGA_GBM_gistic2  _GENOMIC_ID_TCGA_GBM_hMethyl450  \n",
       "sampleID                                                                        \n",
       "TCGA-02-0001-01  TCGA-02-0001-01C-01D-0182-01                              NaN  \n",
       "TCGA-02-0002-01                           NaN                              NaN  \n",
       "TCGA-02-0003-01  TCGA-02-0003-01A-01D-0182-01                              NaN  \n",
       "\n",
       "[3 rows x 128 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical = pd.read_csv('/Users/yuanqizhao/Desktop/CompMed_Project/data/GBM_clinicalMatrix', sep='\\t',header=0,index_col=0)\n",
    "clinical[:3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['CDE_DxAge', 'CDE_alk_chemoradiation_standard',\n",
       "       'CDE_chemo_adjuvant_alk', 'CDE_chemo_adjuvant_tmz', 'CDE_chemo_alk',\n",
       "       'CDE_chemo_alk_days', 'CDE_chemo_alk_long', 'CDE_chemo_tmz',\n",
       "       'CDE_chemo_tmz_days', 'CDE_chemo_tmz_long',\n",
       "       ...\n",
       "       '_GENOMIC_ID_TCGA_GBM_exp_u133a',\n",
       "       '_GENOMIC_ID_TCGA_GBM_exp_HiSeqV2_PANCAN',\n",
       "       '_GENOMIC_ID_TCGA_GBM_RPPA_RBN', '_GENOMIC_ID_TCGA_GBM_gistic2thd',\n",
       "       '_GENOMIC_ID_TCGA_GBM_PDMRNAseqCNV', '_GENOMIC_ID_TCGA_GBM_mutation',\n",
       "       '_GENOMIC_ID_TCGA_GBM_exp_HiSeqV2_percentile',\n",
       "       '_GENOMIC_ID_TCGA_GBM_exp_HiSeqV2_exon', '_GENOMIC_ID_TCGA_GBM_gistic2',\n",
       "       '_GENOMIC_ID_TCGA_GBM_hMethyl450'],\n",
       "      dtype='object', length=128)"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clinical.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Append subtype labels"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = group.merge(clinical,how='inner',left_index=True, right_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subtype</th>\n",
       "      <th>CDE_DxAge</th>\n",
       "      <th>CDE_alk_chemoradiation_standard</th>\n",
       "      <th>CDE_chemo_adjuvant_alk</th>\n",
       "      <th>CDE_chemo_adjuvant_tmz</th>\n",
       "      <th>CDE_chemo_alk</th>\n",
       "      <th>CDE_chemo_alk_days</th>\n",
       "      <th>CDE_chemo_alk_long</th>\n",
       "      <th>CDE_chemo_tmz</th>\n",
       "      <th>CDE_chemo_tmz_days</th>\n",
       "      <th>...</th>\n",
       "      <th>_GENOMIC_ID_TCGA_GBM_exp_u133a</th>\n",
       "      <th>_GENOMIC_ID_TCGA_GBM_exp_HiSeqV2_PANCAN</th>\n",
       "      <th>_GENOMIC_ID_TCGA_GBM_RPPA_RBN</th>\n",
       "      <th>_GENOMIC_ID_TCGA_GBM_gistic2thd</th>\n",
       "      <th>_GENOMIC_ID_TCGA_GBM_PDMRNAseqCNV</th>\n",
       "      <th>_GENOMIC_ID_TCGA_GBM_mutation</th>\n",
       "      <th>_GENOMIC_ID_TCGA_GBM_exp_HiSeqV2_percentile</th>\n",
       "      <th>_GENOMIC_ID_TCGA_GBM_exp_HiSeqV2_exon</th>\n",
       "      <th>_GENOMIC_ID_TCGA_GBM_gistic2</th>\n",
       "      <th>_GENOMIC_ID_TCGA_GBM_hMethyl450</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>TCGA-02-0001-01</td>\n",
       "      <td>3</td>\n",
       "      <td>44.30</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>03c14838-bbab-4487-80c3-c5fdf6ec83f9</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-02-0001-01C-01D-0182-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-02-0001-01C-01D-0182-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TCGA-02-0003-01</td>\n",
       "      <td>3</td>\n",
       "      <td>50.21</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>4298ad72-e8eb-4749-95da-cce602107724</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-02-0003-01A-21-1898-20</td>\n",
       "      <td>TCGA-02-0003-01A-01D-0182-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-02-0003-01A-01D-1490-08</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-02-0003-01A-01D-0182-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TCGA-02-0007-01</td>\n",
       "      <td>1</td>\n",
       "      <td>40.53</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>306.0</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>306.0</td>\n",
       "      <td>...</td>\n",
       "      <td>b972e653-da0c-4db0-9769-e82cc8a408e0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-02-0007-01A-01D-0182-01</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCGA-02-0007-01A-01D-0182-01</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 129 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 subtype  CDE_DxAge CDE_alk_chemoradiation_standard  \\\n",
       "TCGA-02-0001-01        3      44.30                           False   \n",
       "TCGA-02-0003-01        3      50.21                           False   \n",
       "TCGA-02-0007-01        1      40.53                            True   \n",
       "\n",
       "                CDE_chemo_adjuvant_alk CDE_chemo_adjuvant_tmz CDE_chemo_alk  \\\n",
       "TCGA-02-0001-01                  False                  False         False   \n",
       "TCGA-02-0003-01                  False                  False         False   \n",
       "TCGA-02-0007-01                   True                   True          True   \n",
       "\n",
       "                 CDE_chemo_alk_days CDE_chemo_alk_long CDE_chemo_tmz  \\\n",
       "TCGA-02-0001-01                 0.0              False         False   \n",
       "TCGA-02-0003-01                 0.0              False         False   \n",
       "TCGA-02-0007-01               306.0               True          True   \n",
       "\n",
       "                 CDE_chemo_tmz_days  ...  \\\n",
       "TCGA-02-0001-01                 0.0  ...   \n",
       "TCGA-02-0003-01                 0.0  ...   \n",
       "TCGA-02-0007-01               306.0  ...   \n",
       "\n",
       "                       _GENOMIC_ID_TCGA_GBM_exp_u133a  \\\n",
       "TCGA-02-0001-01  03c14838-bbab-4487-80c3-c5fdf6ec83f9   \n",
       "TCGA-02-0003-01  4298ad72-e8eb-4749-95da-cce602107724   \n",
       "TCGA-02-0007-01  b972e653-da0c-4db0-9769-e82cc8a408e0   \n",
       "\n",
       "                _GENOMIC_ID_TCGA_GBM_exp_HiSeqV2_PANCAN  \\\n",
       "TCGA-02-0001-01                                     NaN   \n",
       "TCGA-02-0003-01                                     NaN   \n",
       "TCGA-02-0007-01                                     NaN   \n",
       "\n",
       "                _GENOMIC_ID_TCGA_GBM_RPPA_RBN _GENOMIC_ID_TCGA_GBM_gistic2thd  \\\n",
       "TCGA-02-0001-01                           NaN    TCGA-02-0001-01C-01D-0182-01   \n",
       "TCGA-02-0003-01   TCGA-02-0003-01A-21-1898-20    TCGA-02-0003-01A-01D-0182-01   \n",
       "TCGA-02-0007-01                           NaN    TCGA-02-0007-01A-01D-0182-01   \n",
       "\n",
       "                _GENOMIC_ID_TCGA_GBM_PDMRNAseqCNV  \\\n",
       "TCGA-02-0001-01                               NaN   \n",
       "TCGA-02-0003-01                               NaN   \n",
       "TCGA-02-0007-01                               NaN   \n",
       "\n",
       "                _GENOMIC_ID_TCGA_GBM_mutation  \\\n",
       "TCGA-02-0001-01                           NaN   \n",
       "TCGA-02-0003-01  TCGA-02-0003-01A-01D-1490-08   \n",
       "TCGA-02-0007-01                           NaN   \n",
       "\n",
       "                _GENOMIC_ID_TCGA_GBM_exp_HiSeqV2_percentile  \\\n",
       "TCGA-02-0001-01                                         NaN   \n",
       "TCGA-02-0003-01                                         NaN   \n",
       "TCGA-02-0007-01                                         NaN   \n",
       "\n",
       "                _GENOMIC_ID_TCGA_GBM_exp_HiSeqV2_exon  \\\n",
       "TCGA-02-0001-01                                   NaN   \n",
       "TCGA-02-0003-01                                   NaN   \n",
       "TCGA-02-0007-01                                   NaN   \n",
       "\n",
       "                 _GENOMIC_ID_TCGA_GBM_gistic2 _GENOMIC_ID_TCGA_GBM_hMethyl450  \n",
       "TCGA-02-0001-01  TCGA-02-0001-01C-01D-0182-01                             NaN  \n",
       "TCGA-02-0003-01  TCGA-02-0003-01A-01D-0182-01                             NaN  \n",
       "TCGA-02-0007-01  TCGA-02-0007-01A-01D-0182-01                             NaN  \n",
       "\n",
       "[3 rows x 129 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df.rename(columns={'x':'subtype'})\n",
    "df[:3]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Subtype Clinical Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['subtype',\n",
       " 'CDE_DxAge',\n",
       " 'CDE_alk_chemoradiation_standard',\n",
       " 'CDE_chemo_adjuvant_alk',\n",
       " 'CDE_chemo_adjuvant_tmz',\n",
       " 'CDE_chemo_alk',\n",
       " 'CDE_chemo_alk_days',\n",
       " 'CDE_chemo_alk_long',\n",
       " 'CDE_chemo_tmz',\n",
       " 'CDE_chemo_tmz_days',\n",
       " 'CDE_chemo_tmz_long',\n",
       " 'CDE_missing',\n",
       " 'CDE_missingflag',\n",
       " 'CDE_previously_treated',\n",
       " 'CDE_radiation_adjuvant',\n",
       " 'CDE_radiation_adjuvant_standard',\n",
       " 'CDE_radiation_adjuvant_standard_probable',\n",
       " 'CDE_radiation_any',\n",
       " 'CDE_radiation_standard',\n",
       " 'CDE_radiation_standard_probable',\n",
       " 'CDE_sourcesite',\n",
       " 'CDE_survival_time',\n",
       " 'CDE_suspect',\n",
       " 'CDE_therapy',\n",
       " 'CDE_tmz_chemoradiation_standard',\n",
       " 'CDE_vital_status',\n",
       " 'G_CIMP_STATUS',\n",
       " 'GeneExp_Subtype',\n",
       " 'In_Cancer_Cell_Paper',\n",
       " '_INTEGRATION',\n",
       " '_PANCAN_CNA_PANCAN_K8',\n",
       " '_PANCAN_Cluster_Cluster_PANCAN',\n",
       " '_PANCAN_DNAMethyl_GBM',\n",
       " '_PANCAN_DNAMethyl_PANCAN',\n",
       " '_PANCAN_RPPA_PANCAN_K8',\n",
       " '_PANCAN_UNC_RNAseq_PANCAN_K16',\n",
       " '_PANCAN_mutation_PANCAN',\n",
       " '_PATIENT',\n",
       " '_cohort',\n",
       " '_primary_disease',\n",
       " '_primary_site',\n",
       " 'additional_chemo_therapy',\n",
       " 'additional_drug_therapy',\n",
       " 'additional_immuno_therapy',\n",
       " 'additional_pharmaceutical_therapy',\n",
       " 'additional_radiation_therapy',\n",
       " 'age_at_initial_pathologic_diagnosis',\n",
       " 'anatomic_organ_subdivision',\n",
       " 'bcr_followup_barcode',\n",
       " 'bcr_patient_barcode',\n",
       " 'bcr_patient_uuid',\n",
       " 'bcr_sample_barcode',\n",
       " 'chemo_therapy',\n",
       " 'days_to_birth',\n",
       " 'days_to_collection',\n",
       " 'days_to_death',\n",
       " 'days_to_initial_pathologic_diagnosis',\n",
       " 'days_to_last_followup',\n",
       " 'days_to_new_tumor_event_additional_surgery_procedure',\n",
       " 'days_to_new_tumor_event_after_initial_treatment',\n",
       " 'days_to_tumor_progression',\n",
       " 'days_to_tumor_recurrence',\n",
       " 'eastern_cancer_oncology_group',\n",
       " 'followup_case_report_form_submission_reason',\n",
       " 'followup_treatment_success',\n",
       " 'form_completion_date',\n",
       " 'gender',\n",
       " 'histological_type',\n",
       " 'history_of_neoadjuvant_treatment',\n",
       " 'hormonal_therapy',\n",
       " 'icd_10',\n",
       " 'icd_o_3_histology',\n",
       " 'icd_o_3_site',\n",
       " 'immuno_therapy',\n",
       " 'informed_consent_verified',\n",
       " 'initial_pathologic_diagnosis_method',\n",
       " 'initial_weight',\n",
       " 'intermediate_dimension',\n",
       " 'is_ffpe',\n",
       " 'karnofsky_performance_score',\n",
       " 'longest_dimension',\n",
       " 'lost_follow_up',\n",
       " 'new_neoplasm_event_type',\n",
       " 'new_tumor_event_additional_surgery_procedure',\n",
       " 'new_tumor_event_after_initial_treatment',\n",
       " 'oct_embedded',\n",
       " 'other_dx',\n",
       " 'pathology_report_file_name',\n",
       " 'patient_id',\n",
       " 'performance_status_scale_timing',\n",
       " 'person_neoplasm_cancer_status',\n",
       " 'postoperative_rx_tx',\n",
       " 'pretreatment_history',\n",
       " 'primary_therapy_outcome_success',\n",
       " 'prior_glioma',\n",
       " 'radiation_therapy',\n",
       " 'sample_type',\n",
       " 'sample_type_id',\n",
       " 'shortest_dimension',\n",
       " 'targeted_molecular_therapy',\n",
       " 'tissue_prospective_collection_indicator',\n",
       " 'tissue_retrospective_collection_indicator',\n",
       " 'tissue_source_site',\n",
       " 'tumor_tissue_site',\n",
       " 'vial_number',\n",
       " 'vital_status',\n",
       " 'year_of_initial_pathologic_diagnosis',\n",
       " '_GENOMIC_ID_data/public/TCGA/GBM/miRNA_HiSeq_gene',\n",
       " '_GENOMIC_ID_TCGA_GBM_miRNA_HiSeq',\n",
       " '_GENOMIC_ID_TCGA_GBM_G4502A_07_1',\n",
       " '_GENOMIC_ID_TCGA_GBM_PDMRNAseq',\n",
       " '_GENOMIC_ID_TCGA_GBM_G4502A_07_2',\n",
       " '_GENOMIC_ID_TCGA_GBM_PDMarray',\n",
       " '_GENOMIC_ID_TCGA_GBM_RPPA',\n",
       " '_GENOMIC_ID_TCGA_GBM_hMethyl27',\n",
       " '_GENOMIC_ID_TCGA_GBM_exp_HiSeqV2',\n",
       " '_GENOMIC_ID_TCGA_GBM_mutation_ucsc_maf_gene',\n",
       " '_GENOMIC_ID_TCGA_GBM_PDMarrayCNV',\n",
       " '_GENOMIC_ID_TCGA_GBM_mutation_broad_gene',\n",
       " '_GENOMIC_ID_TCGA_GBM_exp_u133a',\n",
       " '_GENOMIC_ID_TCGA_GBM_exp_HiSeqV2_PANCAN',\n",
       " '_GENOMIC_ID_TCGA_GBM_RPPA_RBN',\n",
       " '_GENOMIC_ID_TCGA_GBM_gistic2thd',\n",
       " '_GENOMIC_ID_TCGA_GBM_PDMRNAseqCNV',\n",
       " '_GENOMIC_ID_TCGA_GBM_mutation',\n",
       " '_GENOMIC_ID_TCGA_GBM_exp_HiSeqV2_percentile',\n",
       " '_GENOMIC_ID_TCGA_GBM_exp_HiSeqV2_exon',\n",
       " '_GENOMIC_ID_TCGA_GBM_gistic2',\n",
       " '_GENOMIC_ID_TCGA_GBM_hMethyl450']"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(df.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "drop_ls = [name for name in df.columns if '_GENOMIC_ID' in name]\n",
    "df = df.drop(columns = drop_ls)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(347, 107)"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(347, 104)"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "drop_ls = [name for name in df.columns if 'id' in name]\n",
    "df = df.drop(columns = drop_ls)\n",
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "subtype                                   int64\n",
       "CDE_DxAge                               float64\n",
       "CDE_alk_chemoradiation_standard          object\n",
       "CDE_chemo_adjuvant_alk                   object\n",
       "CDE_chemo_adjuvant_tmz                   object\n",
       "                                         ...   \n",
       "tissue_source_site                       object\n",
       "tumor_tissue_site                        object\n",
       "vial_number                              object\n",
       "vital_status                             object\n",
       "year_of_initial_pathologic_diagnosis    float64\n",
       "Length: 104, dtype: object"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.dtypes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['CDE_DxAge',\n",
       " 'CDE_chemo_alk_days',\n",
       " 'CDE_chemo_tmz_days',\n",
       " 'CDE_sourcesite',\n",
       " 'CDE_survival_time',\n",
       " 'age_at_initial_pathologic_diagnosis',\n",
       " 'days_to_birth',\n",
       " 'days_to_collection',\n",
       " 'days_to_initial_pathologic_diagnosis',\n",
       " 'days_to_last_followup',\n",
       " 'days_to_new_tumor_event_additional_surgery_procedure',\n",
       " 'days_to_tumor_recurrence',\n",
       " 'eastern_cancer_oncology_group',\n",
       " 'initial_weight',\n",
       " 'intermediate_dimension',\n",
       " 'karnofsky_performance_score',\n",
       " 'longest_dimension',\n",
       " 'shortest_dimension',\n",
       " 'year_of_initial_pathologic_diagnosis']"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "continuous = [name for name in df.columns if df[name].dtype=='float64']\n",
    "continuous"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(347, 19)"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "continuous_df = df[continuous]\n",
    "continuous_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['CDE_alk_chemoradiation_standard',\n",
       " 'CDE_chemo_adjuvant_alk',\n",
       " 'CDE_chemo_adjuvant_tmz',\n",
       " 'CDE_chemo_alk',\n",
       " 'CDE_chemo_alk_long',\n",
       " 'CDE_chemo_tmz',\n",
       " 'CDE_chemo_tmz_long',\n",
       " 'CDE_missing',\n",
       " 'CDE_missingflag',\n",
       " 'CDE_previously_treated',\n",
       " 'CDE_radiation_adjuvant',\n",
       " 'CDE_radiation_adjuvant_standard',\n",
       " 'CDE_radiation_adjuvant_standard_probable',\n",
       " 'CDE_radiation_any',\n",
       " 'CDE_radiation_standard',\n",
       " 'CDE_radiation_standard_probable',\n",
       " 'CDE_suspect',\n",
       " 'CDE_therapy',\n",
       " 'CDE_tmz_chemoradiation_standard',\n",
       " 'CDE_vital_status',\n",
       " 'G_CIMP_STATUS',\n",
       " 'GeneExp_Subtype',\n",
       " 'In_Cancer_Cell_Paper',\n",
       " '_INTEGRATION',\n",
       " '_PANCAN_CNA_PANCAN_K8',\n",
       " '_PANCAN_Cluster_Cluster_PANCAN',\n",
       " '_PANCAN_DNAMethyl_GBM',\n",
       " '_PANCAN_DNAMethyl_PANCAN',\n",
       " '_PANCAN_RPPA_PANCAN_K8',\n",
       " '_PANCAN_UNC_RNAseq_PANCAN_K16',\n",
       " '_PANCAN_mutation_PANCAN',\n",
       " '_PATIENT',\n",
       " '_cohort',\n",
       " '_primary_disease',\n",
       " '_primary_site',\n",
       " 'additional_chemo_therapy',\n",
       " 'additional_drug_therapy',\n",
       " 'additional_immuno_therapy',\n",
       " 'additional_pharmaceutical_therapy',\n",
       " 'additional_radiation_therapy',\n",
       " 'anatomic_organ_subdivision',\n",
       " 'bcr_followup_barcode',\n",
       " 'bcr_patient_barcode',\n",
       " 'bcr_sample_barcode',\n",
       " 'chemo_therapy',\n",
       " 'days_to_death',\n",
       " 'days_to_new_tumor_event_after_initial_treatment',\n",
       " 'days_to_tumor_progression',\n",
       " 'followup_case_report_form_submission_reason',\n",
       " 'followup_treatment_success',\n",
       " 'form_completion_date',\n",
       " 'gender',\n",
       " 'histological_type',\n",
       " 'history_of_neoadjuvant_treatment',\n",
       " 'hormonal_therapy',\n",
       " 'icd_10',\n",
       " 'icd_o_3_histology',\n",
       " 'icd_o_3_site',\n",
       " 'immuno_therapy',\n",
       " 'informed_consent_verified',\n",
       " 'initial_pathologic_diagnosis_method',\n",
       " 'is_ffpe',\n",
       " 'lost_follow_up',\n",
       " 'new_neoplasm_event_type',\n",
       " 'new_tumor_event_additional_surgery_procedure',\n",
       " 'new_tumor_event_after_initial_treatment',\n",
       " 'oct_embedded',\n",
       " 'other_dx',\n",
       " 'pathology_report_file_name',\n",
       " 'performance_status_scale_timing',\n",
       " 'person_neoplasm_cancer_status',\n",
       " 'postoperative_rx_tx',\n",
       " 'pretreatment_history',\n",
       " 'primary_therapy_outcome_success',\n",
       " 'prior_glioma',\n",
       " 'radiation_therapy',\n",
       " 'sample_type',\n",
       " 'targeted_molecular_therapy',\n",
       " 'tissue_prospective_collection_indicator',\n",
       " 'tissue_retrospective_collection_indicator',\n",
       " 'tissue_source_site',\n",
       " 'tumor_tissue_site',\n",
       " 'vial_number',\n",
       " 'vital_status']"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "discrete = [name for name in df.columns if df[name].dtype=='object']\n",
    "discrete"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(347, 84)"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "discrete_df = df[discrete]\n",
    "discrete_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "TCGA-02-0001-01    False\n",
       "TCGA-02-0003-01    False\n",
       "TCGA-02-0007-01     True\n",
       "TCGA-02-0009-01    False\n",
       "TCGA-02-0010-01     True\n",
       "                   ...  \n",
       "TCGA-76-6193-01      NaN\n",
       "TCGA-76-6282-01      NaN\n",
       "TCGA-76-6285-01      NaN\n",
       "TCGA-81-5910-01    False\n",
       "TCGA-87-5896-01    False\n",
       "Name: CDE_chemo_tmz, Length: 347, dtype: object"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "discrete_df['CDE_chemo_tmz']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Differential analysis of subtypes on continuous variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Collecting bioinfokit\n",
      "\u001b[?25l  Downloading https://files.pythonhosted.org/packages/f5/48/d117e32009dfdfbac82d9e17bdcea730b81ba6e137b29f14d59811a7488e/bioinfokit-1.0.4.tar.gz (76kB)\n",
      "\u001b[K     |████████████████████████████████| 81kB 5.3MB/s eta 0:00:011\n",
      "\u001b[?25hRequirement already satisfied: pandas in /Users/yuanqizhao/opt/anaconda3/lib/python3.7/site-packages (from bioinfokit) (0.25.1)\n",
      "Requirement already satisfied: numpy in /Users/yuanqizhao/opt/anaconda3/lib/python3.7/site-packages (from bioinfokit) (1.17.2)\n",
      "Requirement already satisfied: matplotlib in /Users/yuanqizhao/opt/anaconda3/lib/python3.7/site-packages (from bioinfokit) (3.1.3)\n",
      "Requirement already satisfied: scipy in /Users/yuanqizhao/opt/anaconda3/lib/python3.7/site-packages (from bioinfokit) (1.3.1)\n",
      "Requirement already satisfied: scikit-learn in /Users/yuanqizhao/opt/anaconda3/lib/python3.7/site-packages (from bioinfokit) (0.22.2.post1)\n",
      "Requirement already satisfied: seaborn in /Users/yuanqizhao/opt/anaconda3/lib/python3.7/site-packages (from bioinfokit) (0.9.0)\n",
      "Collecting matplotlib_venn (from bioinfokit)\n",
      "  Downloading https://files.pythonhosted.org/packages/ca/e8/53441d7feb29ab10de3bd46c05358c41f0ba2f57395e88ffbb62d6b69df3/matplotlib-venn-0.11.6.tar.gz\n",
      "Collecting tabulate (from bioinfokit)\n",
      "  Downloading https://files.pythonhosted.org/packages/c4/f4/770ae9385990f5a19a91431163d262182d3203662ea2b5739d0fcfc080f1/tabulate-0.8.7-py3-none-any.whl\n",
      "Requirement already satisfied: statsmodels in /Users/yuanqizhao/opt/anaconda3/lib/python3.7/site-packages (from bioinfokit) (0.10.1)\n",
      "Collecting textwrap3 (from bioinfokit)\n",
      "  Downloading https://files.pythonhosted.org/packages/77/9c/a53e561d496ee5866bbeea4d3a850b3b545ed854f8a21007c1e0d872e94d/textwrap3-0.9.2-py2.py3-none-any.whl\n",
      "Collecting adjustText (from bioinfokit)\n",
      "  Downloading https://files.pythonhosted.org/packages/9e/15/4157718bf323fd5f5b81c891c660d0f388e042d2689a558bf1389632dc44/adjustText-0.7.3.tar.gz\n",
      "Requirement already satisfied: pytz>=2017.2 in /Users/yuanqizhao/opt/anaconda3/lib/python3.7/site-packages (from pandas->bioinfokit) (2019.3)\n",
      "Requirement already satisfied: python-dateutil>=2.6.1 in /Users/yuanqizhao/opt/anaconda3/lib/python3.7/site-packages (from pandas->bioinfokit) (2.8.0)\n",
      "Requirement already satisfied: pyparsing!=2.0.4,!=2.1.2,!=2.1.6,>=2.0.1 in /Users/yuanqizhao/opt/anaconda3/lib/python3.7/site-packages (from matplotlib->bioinfokit) (2.4.2)\n",
      "Requirement already satisfied: kiwisolver>=1.0.1 in /Users/yuanqizhao/opt/anaconda3/lib/python3.7/site-packages (from matplotlib->bioinfokit) (1.1.0)\n",
      "Requirement already satisfied: cycler>=0.10 in /Users/yuanqizhao/opt/anaconda3/lib/python3.7/site-packages (from matplotlib->bioinfokit) (0.10.0)\n",
      "Requirement already satisfied: joblib>=0.11 in /Users/yuanqizhao/opt/anaconda3/lib/python3.7/site-packages (from scikit-learn->bioinfokit) (0.13.2)\n",
      "Requirement already satisfied: patsy>=0.4.0 in /Users/yuanqizhao/opt/anaconda3/lib/python3.7/site-packages (from statsmodels->bioinfokit) (0.5.1)\n",
      "Requirement already satisfied: six>=1.5 in /Users/yuanqizhao/opt/anaconda3/lib/python3.7/site-packages (from python-dateutil>=2.6.1->pandas->bioinfokit) (1.12.0)\n",
      "Requirement already satisfied: setuptools in /Users/yuanqizhao/opt/anaconda3/lib/python3.7/site-packages (from kiwisolver>=1.0.1->matplotlib->bioinfokit) (41.4.0)\n",
      "Building wheels for collected packages: bioinfokit, matplotlib-venn, adjustText\n",
      "  Building wheel for bioinfokit (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for bioinfokit: filename=bioinfokit-1.0.4-cp37-none-any.whl size=54151 sha256=b6d2171088a90fa4ace20c48f361f3d107c647944aa79c982e5a3ebb1453a1f8\n",
      "  Stored in directory: /Users/yuanqizhao/Library/Caches/pip/wheels/af/ef/fc/e1c756425926efdbde4628e7b529dabf0820c0e27265613d25\n",
      "  Building wheel for matplotlib-venn (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for matplotlib-venn: filename=matplotlib_venn-0.11.6-cp37-none-any.whl size=32065 sha256=1e6f2d378b2cac3724f628f27e844cc283fc6e40d7bbb7fa7539f80bf3d675f2\n",
      "  Stored in directory: /Users/yuanqizhao/Library/Caches/pip/wheels/c5/9c/ac/77b0c9151c4c264bec6ba18120d73bb24df5b4c1f09c17a41c\n",
      "  Building wheel for adjustText (setup.py) ... \u001b[?25ldone\n",
      "\u001b[?25h  Created wheel for adjustText: filename=adjustText-0.7.3-cp37-none-any.whl size=7099 sha256=550403d3ccba360fd28648ab6af2d572eaf4d8da804c4a9f439063997ebb778e\n",
      "  Stored in directory: /Users/yuanqizhao/Library/Caches/pip/wheels/41/95/74/7d347e136d672f8bc28e937032bc92baf4f80856763a7e7b72\n",
      "Successfully built bioinfokit matplotlib-venn adjustText\n",
      "Installing collected packages: matplotlib-venn, tabulate, textwrap3, adjustText, bioinfokit\n",
      "Successfully installed adjustText-0.7.3 bioinfokit-1.0.4 matplotlib-venn-0.11.6 tabulate-0.8.7 textwrap3-0.9.2\n"
     ]
    }
   ],
   "source": [
    "!pip install bioinfokit"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CDE_DxAge</th>\n",
       "      <th>CDE_chemo_alk_days</th>\n",
       "      <th>CDE_chemo_tmz_days</th>\n",
       "      <th>CDE_sourcesite</th>\n",
       "      <th>CDE_survival_time</th>\n",
       "      <th>age_at_initial_pathologic_diagnosis</th>\n",
       "      <th>days_to_birth</th>\n",
       "      <th>days_to_collection</th>\n",
       "      <th>days_to_initial_pathologic_diagnosis</th>\n",
       "      <th>days_to_last_followup</th>\n",
       "      <th>days_to_new_tumor_event_additional_surgery_procedure</th>\n",
       "      <th>days_to_tumor_recurrence</th>\n",
       "      <th>eastern_cancer_oncology_group</th>\n",
       "      <th>initial_weight</th>\n",
       "      <th>intermediate_dimension</th>\n",
       "      <th>karnofsky_performance_score</th>\n",
       "      <th>longest_dimension</th>\n",
       "      <th>shortest_dimension</th>\n",
       "      <th>year_of_initial_pathologic_diagnosis</th>\n",
       "      <th>subtype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>TCGA-02-0001-01</td>\n",
       "      <td>44.30</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>353.0</td>\n",
       "      <td>44.0</td>\n",
       "      <td>-16179.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>279.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>137.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.7</td>\n",
       "      <td>80.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.2</td>\n",
       "      <td>2002.0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TCGA-02-0003-01</td>\n",
       "      <td>50.21</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>144.0</td>\n",
       "      <td>50.0</td>\n",
       "      <td>-18341.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>144.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>40.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.9</td>\n",
       "      <td>100.0</td>\n",
       "      <td>2.1</td>\n",
       "      <td>0.1</td>\n",
       "      <td>2003.0</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>TCGA-02-0007-01</td>\n",
       "      <td>40.53</td>\n",
       "      <td>306.0</td>\n",
       "      <td>306.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>705.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>-14806.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.0</td>\n",
       "      <td>705.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>80.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2002.0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 CDE_DxAge  CDE_chemo_alk_days  CDE_chemo_tmz_days  \\\n",
       "TCGA-02-0001-01      44.30                 0.0                 0.0   \n",
       "TCGA-02-0003-01      50.21                 0.0                 0.0   \n",
       "TCGA-02-0007-01      40.53               306.0               306.0   \n",
       "\n",
       "                 CDE_sourcesite  CDE_survival_time  \\\n",
       "TCGA-02-0001-01             2.0              353.0   \n",
       "TCGA-02-0003-01             2.0              144.0   \n",
       "TCGA-02-0007-01             2.0              705.0   \n",
       "\n",
       "                 age_at_initial_pathologic_diagnosis  days_to_birth  \\\n",
       "TCGA-02-0001-01                                 44.0       -16179.0   \n",
       "TCGA-02-0003-01                                 50.0       -18341.0   \n",
       "TCGA-02-0007-01                                 40.0       -14806.0   \n",
       "\n",
       "                 days_to_collection  days_to_initial_pathologic_diagnosis  \\\n",
       "TCGA-02-0001-01                 NaN                                   0.0   \n",
       "TCGA-02-0003-01                 NaN                                   0.0   \n",
       "TCGA-02-0007-01                 NaN                                   0.0   \n",
       "\n",
       "                 days_to_last_followup  \\\n",
       "TCGA-02-0001-01                  279.0   \n",
       "TCGA-02-0003-01                  144.0   \n",
       "TCGA-02-0007-01                  705.0   \n",
       "\n",
       "                 days_to_new_tumor_event_additional_surgery_procedure  \\\n",
       "TCGA-02-0001-01                                                NaN      \n",
       "TCGA-02-0003-01                                                NaN      \n",
       "TCGA-02-0007-01                                                NaN      \n",
       "\n",
       "                 days_to_tumor_recurrence  eastern_cancer_oncology_group  \\\n",
       "TCGA-02-0001-01                     137.0                            NaN   \n",
       "TCGA-02-0003-01                      40.0                            NaN   \n",
       "TCGA-02-0007-01                       NaN                            NaN   \n",
       "\n",
       "                 initial_weight  intermediate_dimension  \\\n",
       "TCGA-02-0001-01             NaN                     0.7   \n",
       "TCGA-02-0003-01             NaN                     0.9   \n",
       "TCGA-02-0007-01             NaN                     NaN   \n",
       "\n",
       "                 karnofsky_performance_score  longest_dimension  \\\n",
       "TCGA-02-0001-01                         80.0                2.0   \n",
       "TCGA-02-0003-01                        100.0                2.1   \n",
       "TCGA-02-0007-01                         80.0                NaN   \n",
       "\n",
       "                 shortest_dimension  year_of_initial_pathologic_diagnosis  \\\n",
       "TCGA-02-0001-01                 0.2                                2002.0   \n",
       "TCGA-02-0003-01                 0.1                                2003.0   \n",
       "TCGA-02-0007-01                 NaN                                2002.0   \n",
       "\n",
       "                 subtype  \n",
       "TCGA-02-0001-01        3  \n",
       "TCGA-02-0003-01        3  \n",
       "TCGA-02-0007-01        1  "
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_c = pd.concat([continuous_df,df['subtype']],axis=1)\n",
    "df_c[:3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_d = pd.concat([discrete_df,df['subtype']],axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [],
   "source": [
    "import itertools"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [],
   "source": [
    "val_dict = {}\n",
    "groups = [df_c.groupby(['subtype']).get_group(i) for i in range(1,4)]\n",
    "for name in df_c.columns:\n",
    "    if name == 'subtype':\n",
    "        continue\n",
    "    df_name = [group[name].values for group in groups]\n",
    "    g1_n = ['G1']*df_name[0].shape[0]\n",
    "    g2_n = ['G2']*df_name[1].shape[0]\n",
    "    g3_n = ['G3']*df_name[2].shape[0]\n",
    "    subtype = pd.Series(g1_n+g2_n+g3_n)\n",
    "    df_name = pd.Series(itertools.chain.from_iterable(df_name))\n",
    "    df_name = pd.concat([subtype,df_name],axis=1)\n",
    "    df_name.columns = ['subtype','values']\n",
    "    val_dict[name] = df_name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "from bioinfokit.analys import stat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CDE_DxAge \n",
      "   group1 group2      Diff     Lower      Upper   q-value   p-value\n",
      "0     G1     G2  8.831249  3.897849  13.764649  5.961064  0.001000\n",
      "1     G1     G3  2.983769 -1.253978   7.221516  2.344651  0.223419\n",
      "2     G2     G3  5.847480  0.513526  11.181434  3.650627  0.027681\n",
      "CDE_chemo_alk_days \n",
      "   group1 group2       Diff      Lower       Upper   q-value   p-value\n",
      "0     G1     G2   5.800078 -51.503308   63.103463  0.337056  0.900000\n",
      "1     G1     G3  66.977677  17.754576  116.200777  4.531157  0.004246\n",
      "2     G2     G3  61.177599  -0.778383  123.133581  3.288192  0.053820\n",
      "CDE_survival_time \n",
      "   group1 group2        Diff       Lower       Upper   q-value   p-value\n",
      "0     G1     G2  207.221875   38.746034  375.697716  4.095987  0.011223\n",
      "1     G1     G3  112.787500  -32.420414  257.995414  2.586612  0.161963\n",
      "2     G2     G3  320.009375  137.663669  502.355081  5.844236  0.001000\n",
      "age_at_initial_pathologic_diagnosis \n",
      "   group1 group2      Diff     Lower      Upper   q-value   p-value\n",
      "0     G1     G2  7.908448  3.035577  12.781319  5.403153  0.001000\n",
      "1     G1     G3  2.745111 -1.428734   6.918956  2.189598  0.270198\n",
      "2     G2     G3  5.163337 -0.042656  10.369331  3.301929  0.052464\n",
      "days_to_birth \n",
      "   group1 group2         Diff        Lower        Upper   q-value   p-value\n",
      "0     G1     G2  2893.910304  1117.499661  4670.320947  5.423537  0.001000\n",
      "1     G1     G3  1021.978851  -499.601211  2543.558913  2.236083  0.255498\n",
      "2     G2     G3  1871.931453   -25.919460  3769.782366  3.283740  0.054166\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/yuanqizhao/opt/anaconda3/lib/python3.7/site-packages/statsmodels/base/model.py:1752: ValueWarning: covariance of constraints does not have full rank. The number of constraints is 2, but rank is 0\n",
      "  'rank is %d' % (J, J_), ValueWarning)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "days_to_last_followup \n",
      "   group1 group2        Diff       Lower       Upper   q-value   p-value\n",
      "0     G1     G2  185.727572   21.210444  350.244700  3.758575  0.022388\n",
      "1     G1     G3   91.460967  -50.269388  233.191322  2.148478  0.283637\n",
      "2     G2     G3  277.188539  100.967531  453.409547  5.236915  0.001000\n"
     ]
    }
   ],
   "source": [
    "sig_diff_ls = []\n",
    "for key,dfs in val_dict.items():\n",
    "    \n",
    "    ### deal with nas\n",
    "    if pd.isna(dfs).sum().sum()>= dfs.size*.4:\n",
    "        continue\n",
    "    \n",
    "    res = stat()\n",
    "    res.tukey_hsd(df=dfs, res_var='values', xfac_var='subtype', anova_model='values ~ C(subtype)')\n",
    "    pval = res.tukey_summary['p-value']\n",
    "    if pd.Series.min(pval)<=0.05:\n",
    "        sig_diff_ls.append(key)\n",
    "        print(key,'\\n',res.tukey_summary)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [],
   "source": [
    "import matplotlib.pyplot as plt\n",
    "import numpy as np\n",
    "from lifelines import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[-22474.591463414636, -19580.681159420288, -21452.61261261261]"
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ages = []\n",
    "for g in groups.__iter__():\n",
    "    ages.append(pd.Series.mean(g['days_to_birth']))\n",
    "ages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[-61.574223187437354, -53.645701806630925, -58.77428113044551]"
      ]
     },
     "execution_count": 132,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "avg_ages = [i/365 for i in ages]\n",
    "avg_ages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CDE_DxAge\n",
      "61.37515527950317\n",
      "52.543906250000006\n",
      "58.391386138613875\n",
      "CDE_chemo_alk_days\n",
      "148.98757763975155\n",
      "143.1875\n",
      "82.00990099009901\n",
      "CDE_survival_time\n",
      "426.3875\n",
      "633.609375\n",
      "313.6\n",
      "age_at_initial_pathologic_diagnosis\n",
      "61.02439024390244\n",
      "53.11594202898551\n",
      "58.27927927927928\n",
      "days_to_birth\n",
      "-22474.591463414636\n",
      "-19580.681159420288\n",
      "-21452.61261261261\n",
      "days_to_last_followup\n",
      "420.7361963190184\n",
      "606.463768115942\n",
      "329.27522935779814\n"
     ]
    }
   ],
   "source": [
    "for name in sig_diff_ls:\n",
    "    print(name)\n",
    "    for g in groups.__iter__():\n",
    "        print(pd.Series.mean(g[name]))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Discrete Independence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [],
   "source": [
    "from scipy.stats import chi2_contingency as chi2_contingency"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "metadata": {},
   "outputs": [],
   "source": [
    "groups_d = [df_d.groupby(['subtype']).get_group(i) for i in range(1,4)]\n",
    "name_count = {}\n",
    "for name in df_d.columns:\n",
    "    contingency_t = []\n",
    "    categories = df_d[name].unique()\n",
    "    for group in groups_d.__iter__():\n",
    "        counts = []\n",
    "        for c in categories:\n",
    "            counts.append(group[name][group[name]==c].count())\n",
    "        contingency_t.append(pd.Series(counts))\n",
    "    contingency_t = pd.DataFrame(contingency_t)\n",
    "    idx = ['G1','G2','G3']\n",
    "    contingency_t.index = idx\n",
    "    contingency_t.columns = categories\n",
    "    name_count[name] = contingency_t"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>False</th>\n",
       "      <th>True</th>\n",
       "      <th>NaN</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>G1</td>\n",
       "      <td>89</td>\n",
       "      <td>72</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>G2</td>\n",
       "      <td>30</td>\n",
       "      <td>34</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>G3</td>\n",
       "      <td>58</td>\n",
       "      <td>43</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    False  True  NaN\n",
       "G1     89    72    0\n",
       "G2     30    34    0\n",
       "G3     58    43    0"
      ]
     },
     "execution_count": 200,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_count['CDE_alk_chemoradiation_standard']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'CDE_alk_chemoradiation_standard':     False  True \n",
       " G1     89     72\n",
       " G2     30     34\n",
       " G3     58     43, 'CDE_chemo_adjuvant_alk':     False  True \n",
       " G1     50    111\n",
       " G2     21     43\n",
       " G3     31     70, 'CDE_chemo_adjuvant_tmz':     False  True \n",
       " G1     63     98\n",
       " G2     24     40\n",
       " G3     35     66, 'CDE_chemo_alk':     False  True \n",
       " G1     39    122\n",
       " G2     16     48\n",
       " G3     29     72, 'CDE_chemo_alk_long':     False  True \n",
       " G1     44    117\n",
       " G2     20     44\n",
       " G3     37     64, 'CDE_chemo_tmz':     False  True \n",
       " G1     53    108\n",
       " G2     19     45\n",
       " G3     31     70, 'CDE_chemo_tmz_long':     False  True \n",
       " G1     54    107\n",
       " G2     23     41\n",
       " G3     38     63, 'CDE_missing':     histological_type,  vital_status,\n",
       " G1                   0              1\n",
       " G2                   1              0\n",
       " G3                   0              1, 'CDE_missingflag':     False  True \n",
       " G1    160      1\n",
       " G2     63      1\n",
       " G3    100      1, 'CDE_previously_treated':     False  True \n",
       " G1    154      7\n",
       " G2     57      6\n",
       " G3     96      5, 'CDE_radiation_adjuvant':     True   False\n",
       " G1    121     40\n",
       " G2     56      8\n",
       " G3     82     19, 'CDE_radiation_adjuvant_standard':     True   False\n",
       " G1    100     61\n",
       " G2     46     18\n",
       " G3     58     43, 'CDE_radiation_adjuvant_standard_probable':     True   False\n",
       " G1    105     56\n",
       " G2     50     14\n",
       " G3     61     40, 'CDE_radiation_any':     True   False\n",
       " G1    127     34\n",
       " G2     56      8\n",
       " G3     83     18, 'CDE_radiation_standard':     True   False\n",
       " G1    104     57\n",
       " G2     46     18\n",
       " G3     59     42, 'CDE_radiation_standard_probable':     True   False\n",
       " G1    109     52\n",
       " G2     50     14\n",
       " G3     62     39, 'CDE_suspect': Empty DataFrame\n",
       " Columns: []\n",
       " Index: [], 'CDE_therapy':     StandardRadiation  TMZChemoradiation.TMZChemoLong  NonstandardRadiation  \\\n",
       " G1                  8                              72                     5   \n",
       " G2                  7                              32                     2   \n",
       " G3                  3                              43                    11   \n",
       " \n",
       "     StandardRadiation.TMZChemoLong  StandardRadiation.TMZChemo  \\\n",
       " G1                              24                           0   \n",
       " G2                               8                           3   \n",
       " G3                              14                           3   \n",
       " \n",
       "     StandardRadiation.AlkChemoLong  TMZChemoLong  \\\n",
       " G1                               7             4   \n",
       " G2                               1             1   \n",
       " G3                               1             2   \n",
       " \n",
       "     NonstandardRadiation.AlkChemoLong  NonstandardRadiation.TMZChemoLong  \\\n",
       " G1                                  2                                  7   \n",
       " G2                                  0                                  0   \n",
       " G3                                  0                                  4   \n",
       " \n",
       "     NonstandardRadiation.TMZChemo  AlkChemoradiation.AlkChemoLong  AlkChemo  \\\n",
       " G1                              1                               0         3   \n",
       " G2                              1                               2         0   \n",
       " G3                              4                               0         1   \n",
       " \n",
       "     StandardRadiation.AlkChemo  AlkChemoLong  \n",
       " G1                           1             1  \n",
       " G2                           0             0  \n",
       " G3                           0             0  , 'CDE_tmz_chemoradiation_standard':     False  True \n",
       " G1     89     72\n",
       " G2     32     32\n",
       " G3     58     43, 'CDE_vital_status':     DECEASED  LIVING\n",
       " G1       124      36\n",
       " G2        40      24\n",
       " G3        67      33, 'G_CIMP_STATUS':     NON G-CIMP  G-CIMP\n",
       " G1         161       4\n",
       " G2          42      24\n",
       " G3         108       5, 'GeneExp_Subtype':     Classical  Proneural  Mesenchymal  Neural\n",
       " G1         68         25           38      34\n",
       " G2          4         48           12       5\n",
       " G3         23         18           55      17, 'In_Cancer_Cell_Paper':     True   False\n",
       " G1     21    144\n",
       " G2     13     56\n",
       " G3     24     89, '_INTEGRATION':     TCGA-02-0001-01  TCGA-02-0003-01  TCGA-02-0007-01  TCGA-02-0009-01  \\\n",
       " G1                0                0                1                1   \n",
       " G2                0                0                0                0   \n",
       " G3                1                1                0                0   \n",
       " \n",
       "     TCGA-02-0010-01  TCGA-02-0011-01  TCGA-02-0014-01  TCGA-02-0021-01  \\\n",
       " G1                0                0                0                1   \n",
       " G2                1                1                1                0   \n",
       " G3                0                0                0                0   \n",
       " \n",
       "     TCGA-02-0024-01  TCGA-02-0027-01  ...  TCGA-76-4931-01  TCGA-76-4934-01  \\\n",
       " G1                0                0  ...                1                0   \n",
       " G2                1                0  ...                0                1   \n",
       " G3                0                1  ...                0                0   \n",
       " \n",
       "     TCGA-76-4935-01  TCGA-76-6191-01  TCGA-76-6192-01  TCGA-76-6193-01  \\\n",
       " G1                1                0                0                0   \n",
       " G2                0                0                0                0   \n",
       " G3                0                1                1                1   \n",
       " \n",
       "     TCGA-76-6282-01  TCGA-76-6285-01  TCGA-81-5910-01  TCGA-87-5896-01  \n",
       " G1                0                0                0                1  \n",
       " G2                0                0                0                0  \n",
       " G3                1                1                1                0  \n",
       " \n",
       " [3 rows x 347 columns], '_PANCAN_CNA_PANCAN_K8':     High  GBM  Kirc+  Squamous  Iq  BRCA-LUAD+  Quiet  COAD-READ\n",
       " G1     0  143     11         0   1          10      0          0\n",
       " G2     4   25     22         8   1           8      0          1\n",
       " G3     2   80     18         0   0           7      5          1, '_PANCAN_Cluster_Cluster_PANCAN':     C10-GBM  C1-LUAD-enriched  C5-KIRC\n",
       " G1       69                 0        0\n",
       " G2       26                 2        0\n",
       " G3       57                 1        2, '_PANCAN_DNAMethyl_GBM':     cluster 1  cluster 3  cluster 4  cluster 5  cluster 6  cluster 2\n",
       " G1         12         66         36          3         13         31\n",
       " G2          3         12          8         21         19          6\n",
       " G3         35         15          6          4          6         46, '_PANCAN_DNAMethyl_PANCAN':     GBM non-CIMP c15  GBM CIMP c16  Low purity c2  Cluster 5\n",
       " G1               158             3              0          0\n",
       " G2                46            21              0          2\n",
       " G3                97             4              9          1, '_PANCAN_RPPA_PANCAN_K8':     KIRC-like c1  GBM-like c2  squamous c3\n",
       " G1             3           60            2\n",
       " G2             2           26            1\n",
       " G3             8           40            1, '_PANCAN_UNC_RNAseq_PANCAN_K16':     GBM-like c6\n",
       " G1           53\n",
       " G2           23\n",
       " G3           42, '_PANCAN_mutation_PANCAN':     mutation cluster 8  mutation cluster 3  mutation cluster 4  \\\n",
       " G1                   5                   1                   3   \n",
       " G2                   1                   4                   2   \n",
       " G3                   7                   4                   8   \n",
       " \n",
       "     mutation c1uster 9  mutation cluster 5  mutation c1uster 10  \\\n",
       " G1                  27                  10                    1   \n",
       " G2                   5                  11                    1   \n",
       " G3                  27                   4                    2   \n",
       " \n",
       "     mutation cluster 14  mutation cluster 12  mutation cluster 7  \\\n",
       " G1                   27                    4                   2   \n",
       " G2                    2                    1                   1   \n",
       " G3                   19                    0                   2   \n",
       " \n",
       "     mutation cluster 13  mutation cluster 11  \n",
       " G1                    1                    1  \n",
       " G2                    2                    1  \n",
       " G3                    2                    1  , '_PATIENT':     TCGA-02-0001  TCGA-02-0003  TCGA-02-0007  TCGA-02-0009  TCGA-02-0010  \\\n",
       " G1             0             0             1             1             0   \n",
       " G2             0             0             0             0             1   \n",
       " G3             1             1             0             0             0   \n",
       " \n",
       "     TCGA-02-0011  TCGA-02-0014  TCGA-02-0021  TCGA-02-0024  TCGA-02-0027  ...  \\\n",
       " G1             0             0             1             0             0  ...   \n",
       " G2             1             1             0             1             0  ...   \n",
       " G3             0             0             0             0             1  ...   \n",
       " \n",
       "     TCGA-76-4931  TCGA-76-4934  TCGA-76-4935  TCGA-76-6191  TCGA-76-6192  \\\n",
       " G1             1             0             1             0             0   \n",
       " G2             0             1             0             0             0   \n",
       " G3             0             0             0             1             1   \n",
       " \n",
       "     TCGA-76-6193  TCGA-76-6282  TCGA-76-6285  TCGA-81-5910  TCGA-87-5896  \n",
       " G1             0             0             0             0             1  \n",
       " G2             0             0             0             0             0  \n",
       " G3             1             1             1             1             0  \n",
       " \n",
       " [3 rows x 347 columns], '_cohort':     TCGA Glioblastoma (GBM)\n",
       " G1                      165\n",
       " G2                       69\n",
       " G3                      113, '_primary_disease':     glioblastoma multiforme\n",
       " G1                      165\n",
       " G2                       69\n",
       " G3                      113, '_primary_site':     Brain\n",
       " G1    165\n",
       " G2     69\n",
       " G3    113, 'additional_chemo_therapy':     NO  YES\n",
       " G1  87   60\n",
       " G2  43   17\n",
       " G3  60   27, 'additional_drug_therapy':      NO  YES\n",
       " G1  126   24\n",
       " G2   55    6\n",
       " G3   84    7, 'additional_immuno_therapy':      NO  YES\n",
       " G1  148    2\n",
       " G2   60    1\n",
       " G3   91    0, 'additional_pharmaceutical_therapy':     NO  YES\n",
       " G1  85   67\n",
       " G2  41   21\n",
       " G3  65   32, 'additional_radiation_therapy':      NO  YES\n",
       " G1  133   20\n",
       " G2   58    4\n",
       " G3   82   14, 'anatomic_organ_subdivision':     Brain\n",
       " G1    159\n",
       " G2     62\n",
       " G3    101, 'bcr_followup_barcode':     TCGA-02-0001-F22169  TCGA-02-0003-F22176  TCGA-02-0007-F22191  \\\n",
       " G1                    0                    0                    1   \n",
       " G2                    0                    0                    0   \n",
       " G3                    1                    1                    0   \n",
       " \n",
       "     TCGA-02-0009-F22196  TCGA-02-0010-F22198  TCGA-02-0011-F22204  \\\n",
       " G1                    1                    0                    0   \n",
       " G2                    0                    1                    1   \n",
       " G3                    0                    0                    0   \n",
       " \n",
       "     TCGA-02-0014-F22211  TCGA-02-0021-F22225  TCGA-02-0024-F22238  \\\n",
       " G1                    0                    1                    0   \n",
       " G2                    1                    0                    1   \n",
       " G3                    0                    0                    0   \n",
       " \n",
       "     TCGA-02-0027-F22254  ...  TCGA-76-4931-F28535  TCGA-76-4934-F28543  \\\n",
       " G1                    0  ...                    1                    0   \n",
       " G2                    0  ...                    0                    1   \n",
       " G3                    1  ...                    0                    0   \n",
       " \n",
       "     TCGA-76-4935-F28545  TCGA-76-6191-F28550  TCGA-76-6192-F28557  \\\n",
       " G1                    1                    0                    0   \n",
       " G2                    0                    0                    0   \n",
       " G3                    0                    1                    1   \n",
       " \n",
       "     TCGA-76-6193-F28561  TCGA-76-6282-F28565  TCGA-76-6285-F28570  \\\n",
       " G1                    0                    0                    0   \n",
       " G2                    0                    0                    0   \n",
       " G3                    1                    1                    1   \n",
       " \n",
       "     TCGA-81-5910-F28600  TCGA-87-5896-F43207  \n",
       " G1                    0                    1  \n",
       " G2                    0                    0  \n",
       " G3                    1                    0  \n",
       " \n",
       " [3 rows x 340 columns], 'bcr_patient_barcode':     TCGA-02-0001  TCGA-02-0003  TCGA-02-0007  TCGA-02-0009  TCGA-02-0010  \\\n",
       " G1             0             0             1             1             0   \n",
       " G2             0             0             0             0             1   \n",
       " G3             1             1             0             0             0   \n",
       " \n",
       "     TCGA-02-0011  TCGA-02-0014  TCGA-02-0021  TCGA-02-0024  TCGA-02-0027  ...  \\\n",
       " G1             0             0             1             0             0  ...   \n",
       " G2             1             1             0             1             0  ...   \n",
       " G3             0             0             0             0             1  ...   \n",
       " \n",
       "     TCGA-76-4931  TCGA-76-4934  TCGA-76-4935  TCGA-76-6191  TCGA-76-6192  \\\n",
       " G1             1             0             1             0             0   \n",
       " G2             0             1             0             0             0   \n",
       " G3             0             0             0             1             1   \n",
       " \n",
       "     TCGA-76-6193  TCGA-76-6282  TCGA-76-6285  TCGA-81-5910  TCGA-87-5896  \n",
       " G1             0             0             0             0             1  \n",
       " G2             0             0             0             0             0  \n",
       " G3             1             1             1             1             0  \n",
       " \n",
       " [3 rows x 344 columns], 'bcr_sample_barcode':     TCGA-02-0001-01C  TCGA-02-0003-01A  TCGA-02-0009-01A  TCGA-02-0010-01A  \\\n",
       " G1                 0                 0                 1                 0   \n",
       " G2                 0                 0                 0                 1   \n",
       " G3                 1                 1                 0                 0   \n",
       " \n",
       "     TCGA-02-0011-01B  TCGA-02-0021-01A  TCGA-02-0024-01B  TCGA-02-0027-01A  \\\n",
       " G1                 0                 1                 0                 0   \n",
       " G2                 1                 0                 1                 0   \n",
       " G3                 0                 0                 0                 1   \n",
       " \n",
       "     TCGA-02-0033-01A  TCGA-02-0034-01A  ...  TCGA-76-4929-01A  \\\n",
       " G1                 0                 0  ...                 0   \n",
       " G2                 0                 0  ...                 1   \n",
       " G3                 1                 1  ...                 0   \n",
       " \n",
       "     TCGA-76-4931-01A  TCGA-76-4934-01A  TCGA-76-6191-01A  TCGA-76-6192-01A  \\\n",
       " G1                 1                 0                 0                 0   \n",
       " G2                 0                 1                 0                 0   \n",
       " G3                 0                 0                 1                 1   \n",
       " \n",
       "     TCGA-76-6193-01A  TCGA-76-6282-01A  TCGA-76-6285-01A  TCGA-81-5910-01A  \\\n",
       " G1                 0                 0                 0                 0   \n",
       " G2                 0                 0                 0                 0   \n",
       " G3                 1                 1                 1                 1   \n",
       " \n",
       "     TCGA-87-5896-01A  \n",
       " G1                 1  \n",
       " G2                 0  \n",
       " G3                 0  \n",
       " \n",
       " [3 rows x 279 columns], 'chemo_therapy':     YES  NO\n",
       " G1  124  25\n",
       " G2   53   9\n",
       " G3   78  17, 'days_to_death':     358  144  705  322  1077  630  2512  2362  1615  370  ...  290  135  146  \\\n",
       " G1    0    0    1    1     0    0     0     1     0    0  ...    1    1    0   \n",
       " G2    0    0    0    0     1    1     1     0     1    0  ...    0    0    1   \n",
       " G3    1    1    0    0     0    0     0     0     0    1  ...    0    0    0   \n",
       " \n",
       "     94  279  1121  508  82  519  49  \n",
       " G1   0    1     1    0   0    0   0  \n",
       " G2   0    0     0    0   0    0   0  \n",
       " G3   1    0     0    1   1    1   1  \n",
       " \n",
       " [3 rows x 237 columns], 'days_to_new_tumor_event_after_initial_treatment':     137  40  518  264  351  145  2243  255  1401  257  ...  26  122  210  34  \\\n",
       " G1    0   0    2    1    1    0     0    1     0    0  ...   0    1    0   0   \n",
       " G2    0   0    0    0    1    1     1    0     1    0  ...   1    0    1   0   \n",
       " G3    1   1    0    0    1    2     0    0     0    1  ...   0    0    0   1   \n",
       " \n",
       "     112  [Discrepancy]  323  76  64  458  \n",
       " G1    1              1    0   0   0    0  \n",
       " G2    0              0    0   0   0    0  \n",
       " G3    0              0    1   1   1    1  \n",
       " \n",
       " [3 rows x 175 columns], 'days_to_tumor_progression':     518  264  351  144  2243  254  257  1921  32  386  ...  265  233  488  \\\n",
       " G1    2    1    1    0     0    1    0     0   0    0  ...    0    0    0   \n",
       " G2    0    0    1    1     1    0    0     1   0    0  ...    0    0    0   \n",
       " G3    0    0    0    0     0    0    1     0   1    1  ...    1    1    1   \n",
       " \n",
       "     131  181  118  26  122  210  34  \n",
       " G1    2    1    0   0    1    0   0  \n",
       " G2    0    0    1   1    0    1   0  \n",
       " G3    0    0    0   0    0    0   1  \n",
       " \n",
       " [3 rows x 117 columns], 'followup_case_report_form_submission_reason':     Scheduled Follow-up Submission  Additional New Tumor Event\n",
       " G1                              14                           1\n",
       " G2                               6                           0\n",
       " G3                               6                           0, 'followup_treatment_success':     Progressive Disease  Complete Remission/Response\n",
       " G1                    3                            1\n",
       " G2                    1                            0\n",
       " G3                    3                            0, 'form_completion_date':     2008-12-16  2008-12-17  2009-4-6  2010-5-6  2008-12-10  2008-12-8  \\\n",
       " G1           9           2         0         2           6          1   \n",
       " G2           7           4         0         2           1          1   \n",
       " G3          10           6         1         1           6          1   \n",
       " \n",
       "     2013-11-26  2013-2-12  2008-12-11  2009-5-11  ...  2012-4-3  2010-6-3  \\\n",
       " G1           1          1           2          3  ...         1         2   \n",
       " G2           0          0           0          4  ...         1         1   \n",
       " G3           0          1           1          0  ...         0         1   \n",
       " \n",
       "     2010-10-26  2011-5-5  2011-7-21  2011-4-6  2015-3-26  2011-9-22  \\\n",
       " G1           2         0          0         1          1          0   \n",
       " G2           0         0          1         3          0          0   \n",
       " G3           1         1          0         2          0          4   \n",
       " \n",
       "     2011-10-21  2011-6-7  \n",
       " G1           0         0  \n",
       " G2           0         0  \n",
       " G3           1         1  \n",
       " \n",
       " [3 rows x 99 columns], 'gender':     FEMALE  MALE\n",
       " G1      68    96\n",
       " G2      35    34\n",
       " G3      40    71, 'histological_type':     Untreated primary (de novo) GBM  Treated primary GBM  \\\n",
       " G1                              156                    7   \n",
       " G2                               63                    6   \n",
       " G3                              104                    5   \n",
       " \n",
       "     Glioblastoma Multiforme (GBM)  \n",
       " G1                              1  \n",
       " G2                              0  \n",
       " G3                              2  , 'history_of_neoadjuvant_treatment':     Yes   No\n",
       " G1    8  156\n",
       " G2    5   64\n",
       " G3    6  105, 'hormonal_therapy':      NO  YES\n",
       " G1  115   38\n",
       " G2   48   14\n",
       " G3   87    8, 'icd_10':     C71.9  C71.2\n",
       " G1    164      0\n",
       " G2     69      0\n",
       " G3    110      1, 'icd_o_3_histology':     9440/3\n",
       " G1     164\n",
       " G2      69\n",
       " G3     111, 'icd_o_3_site':     C71.9  C71.2\n",
       " G1    164      0\n",
       " G2     69      0\n",
       " G3    110      1, 'immuno_therapy':      NO  YES\n",
       " G1  147    6\n",
       " G2   60    2\n",
       " G3   92    3, 'informed_consent_verified':     YES\n",
       " G1  164\n",
       " G2   69\n",
       " G3  111, 'initial_pathologic_diagnosis_method':     Tumor resection  Excisional Biopsy  Incisional Biopsy  \\\n",
       " G1              131                 28                  2   \n",
       " G2               64                  4                  1   \n",
       " G3              104                  5                  1   \n",
       " \n",
       "     Fine needle aspiration biopsy  Other method, specify:  \n",
       " G1                              2                       1  \n",
       " G2                              0                       0  \n",
       " G3                              1                       0  , 'is_ffpe':      NO\n",
       " G1  128\n",
       " G2   57\n",
       " G3   94, 'lost_follow_up':     YES  NO\n",
       " G1    1   9\n",
       " G2    0   3\n",
       " G3    0   4, 'new_neoplasm_event_type':     Recurrence  Progression of Disease  Locoregional Disease\n",
       " G1          37                      70                     1\n",
       " G2          11                      28                     2\n",
       " G3          19                      45                     2, 'new_tumor_event_additional_surgery_procedure':     YES\n",
       " G1    1\n",
       " G2    2\n",
       " G3    2, 'new_tumor_event_after_initial_treatment':     NO  YES\n",
       " G1   3    0\n",
       " G2   0    1\n",
       " G3   1    2, 'oct_embedded': Empty DataFrame\n",
       " Columns: []\n",
       " Index: [], 'other_dx': Empty DataFrame\n",
       " Columns: []\n",
       " Index: [], 'pathology_report_file_name':     TCGA-02-0001.392585DD-C29D-4226-B34D-1CAB66C10009.pdf  \\\n",
       " G1                                                  0       \n",
       " G2                                                  0       \n",
       " G3                                                  1       \n",
       " \n",
       "     TCGA-02-0003.1E5D6194-FE78-41E6-A05E-92B2A8E0B162.pdf  \\\n",
       " G1                                                  0       \n",
       " G2                                                  0       \n",
       " G3                                                  1       \n",
       " \n",
       "     TCGA-02-0009.1AD9AD8E-0DA4-4D15-B984-A59D88014D51.pdf  \\\n",
       " G1                                                  1       \n",
       " G2                                                  0       \n",
       " G3                                                  0       \n",
       " \n",
       "     TCGA-02-0010.C7B6B2FC-48A7-4A56-893F-F8EDAFCE952A.pdf  \\\n",
       " G1                                                  0       \n",
       " G2                                                  1       \n",
       " G3                                                  0       \n",
       " \n",
       "     TCGA-02-0011.49241786-AEE6-4404-B696-D1AB8BAEE6F4.pdf  \\\n",
       " G1                                                  0       \n",
       " G2                                                  1       \n",
       " G3                                                  0       \n",
       " \n",
       "     TCGA-02-0021.1BEE82DA-EB4A-492E-B872-632F81F7C84E.pdf  \\\n",
       " G1                                                  1       \n",
       " G2                                                  0       \n",
       " G3                                                  0       \n",
       " \n",
       "     TCGA-02-0024.4F8AA412-3AAE-47F2-9F07-F4A9B2EE11DF.pdf  \\\n",
       " G1                                                  0       \n",
       " G2                                                  1       \n",
       " G3                                                  0       \n",
       " \n",
       "     TCGA-02-0027.2E9F38EF-D6A4-4431-8514-08FEFF62AA68.pdf  \\\n",
       " G1                                                  0       \n",
       " G2                                                  0       \n",
       " G3                                                  1       \n",
       " \n",
       "     TCGA-02-0033.BD51E951-1FAF-4FCA-8985-CF63B76411F6.pdf  \\\n",
       " G1                                                  0       \n",
       " G2                                                  0       \n",
       " G3                                                  1       \n",
       " \n",
       "     TCGA-02-0034.834B13FD-1F37-4B25-AD92-90A62D6495B8.pdf  ...  \\\n",
       " G1                                                  0      ...   \n",
       " G2                                                  0      ...   \n",
       " G3                                                  1      ...   \n",
       " \n",
       "     TCGA-76-4929.65054ac8-f9aa-4342-9f18-81c15efd6d81.pdf  \\\n",
       " G1                                                  0       \n",
       " G2                                                  1       \n",
       " G3                                                  0       \n",
       " \n",
       "     TCGA-76-4931.f34d64d6-6103-407f-b45b-4e660a1bfcd1.pdf  \\\n",
       " G1                                                  1       \n",
       " G2                                                  0       \n",
       " G3                                                  0       \n",
       " \n",
       "     TCGA-76-4934.31727bd3-2af9-4648-b5ae-60af7be49ea7.pdf  \\\n",
       " G1                                                  0       \n",
       " G2                                                  1       \n",
       " G3                                                  0       \n",
       " \n",
       "     TCGA-76-6191.d91aa8ca-fe4b-4f2e-89a7-c56d6fe176e1.pdf  \\\n",
       " G1                                                  0       \n",
       " G2                                                  0       \n",
       " G3                                                  1       \n",
       " \n",
       "     TCGA-76-6192.d2a94594-7f5a-40f2-87fa-beee3fa5be5b.pdf  \\\n",
       " G1                                                  0       \n",
       " G2                                                  0       \n",
       " G3                                                  1       \n",
       " \n",
       "     TCGA-76-6193.1ee55ecb-8de5-4a21-9fa0-babcb2d07212.pdf  \\\n",
       " G1                                                  0       \n",
       " G2                                                  0       \n",
       " G3                                                  1       \n",
       " \n",
       "     TCGA-76-6282.fe3b0aee-6477-48dd-af17-54f991fa47a5.pdf  \\\n",
       " G1                                                  0       \n",
       " G2                                                  0       \n",
       " G3                                                  1       \n",
       " \n",
       "     TCGA-76-6285.9b0a966e-0cd8-44a2-8f15-c5f5ed40c189.pdf  \\\n",
       " G1                                                  0       \n",
       " G2                                                  0       \n",
       " G3                                                  1       \n",
       " \n",
       "     TCGA-81-5910.9d6e4a44-727e-4713-842e-a75ded996614.pdf  \\\n",
       " G1                                                  0       \n",
       " G2                                                  0       \n",
       " G3                                                  1       \n",
       " \n",
       "     TCGA-87-5896.a743b83f-a4ff-4b06-9728-765a6bb6807f.pdf  \n",
       " G1                                                  1      \n",
       " G2                                                  0      \n",
       " G3                                                  0      \n",
       " \n",
       " [3 rows x 279 columns], 'performance_status_scale_timing':     Post-Adjuvant Therapy  Pre-Adjuvant Therapy  Preoperative  Other\n",
       " G1                     29                    51            33      4\n",
       " G2                     10                    18            10      6\n",
       " G3                     20                    24            16      2, 'person_neoplasm_cancer_status':     WITH TUMOR  TUMOR FREE\n",
       " G1         138           8\n",
       " G2          50           6\n",
       " G3          96           5, 'postoperative_rx_tx':     YES  NO\n",
       " G1  134  16\n",
       " G2   59   5\n",
       " G3   89  16, 'pretreatment_history':     YES   NO  [Discrepancy]\n",
       " G1    8  152              1\n",
       " G2    5   59              0\n",
       " G3    6   95              0, 'primary_therapy_outcome_success':     Complete Remission/Response  Progressive Disease  Stable Disease\n",
       " G1                            1                    3               0\n",
       " G2                            0                    0               1\n",
       " G3                            1                    2               2, 'prior_glioma':      NO  YES\n",
       " G1  161    3\n",
       " G2   66    3\n",
       " G3  107    4, 'radiation_therapy':     YES  NO\n",
       " G1  133  23\n",
       " G2   60   7\n",
       " G3   93  14, 'sample_type':     Primary Tumor\n",
       " G1            165\n",
       " G2             69\n",
       " G3            113, 'targeted_molecular_therapy':      NO  YES\n",
       " G1  126   25\n",
       " G2   56    6\n",
       " G3   85    8, 'tissue_prospective_collection_indicator': Empty DataFrame\n",
       " Columns: []\n",
       " Index: [], 'tissue_retrospective_collection_indicator': Empty DataFrame\n",
       " Columns: []\n",
       " Index: [], 'tissue_source_site':     02  06  12  14  15  16  19  26  27  28  32  41  76  81  87\n",
       " G1  13  29  31  20   1   8  18   7  14   6  10   4   2   0   1\n",
       " G2  13  11   3  12   2   3   5   2   2   7   4   2   3   0   0\n",
       " G3  18  29   3   6   1   2  12   3   1  14  11   3   7   1   0, 'tumor_tissue_site':     Brain\n",
       " G1    164\n",
       " G2     69\n",
       " G3    111, 'vial_number':     C    A   B\n",
       " G1  1  115  12\n",
       " G2  2   51   4\n",
       " G3  2   87   5, 'vital_status':     DECEASED  LIVING\n",
       " G1       138      25\n",
       " G2        48      21\n",
       " G3        84      27, 'subtype':       3    1   2\n",
       " G1    0  165   0\n",
       " G2    0    0  69\n",
       " G3  113    0   0, 'CDE_survival_time':     353.0   144.0   705.0   322.0   1077.0  630.0   2511.0  2361.0  1614.0  \\\n",
       " G1       0       0       1       1       0       0       0       1       0   \n",
       " G2       0       0       0       0       1       1       1       0       1   \n",
       " G3       1       1       0       0       0       0       0       0       0   \n",
       " \n",
       "     370.0   ...  330.0   26.0    406.0   272.0   290.0   135.0   351.0   \\\n",
       " G1       0  ...       0       0       1       0       1       1       0   \n",
       " G2       0  ...       0       1       0       0       0       0       1   \n",
       " G3       1  ...       1       0       0       1       0       0       0   \n",
       " \n",
       "     94.0    49.0    2.0     \n",
       " G1       0       0       1  \n",
       " G2       0       0       0  \n",
       " G3       1       1       0  \n",
       " \n",
       " [3 rows x 276 columns]}"
      ]
     },
     "execution_count": 212,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for key, value in name_count.items():\n",
    "    ls1 = value.sum(axis =1)\n",
    "#     print('h',ls1)\n",
    "    idx1 = np.where(np.asarray(ls1)==0)[0]\n",
    "#     print('d',idx1)\n",
    "    if len(idx1)!=0:\n",
    "        name_count[key] = pd.DataFrame([])\n",
    "        continue\n",
    "#     print(key,name_count[key])\n",
    "    ls = value.sum(axis=0)\n",
    "    idx = np.where(np.asarray(ls)==0)[0]\n",
    "    if len(idx)!=0:\n",
    "        drop_col = [value.columns[i] for i in idx]\n",
    "        new_df = value.drop(drop_col,axis=1)\n",
    "        name_count[key] = new_df\n",
    "name_count    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    False  True \n",
      "G1     89     72\n",
      "G2     30     34\n",
      "G3     58     43\n",
      "    False  True \n",
      "G1     50    111\n",
      "G2     21     43\n",
      "G3     31     70\n",
      "    False  True \n",
      "G1     63     98\n",
      "G2     24     40\n",
      "G3     35     66\n",
      "    False  True \n",
      "G1     39    122\n",
      "G2     16     48\n",
      "G3     29     72\n",
      "    False  True \n",
      "G1     44    117\n",
      "G2     20     44\n",
      "G3     37     64\n",
      "    False  True \n",
      "G1     53    108\n",
      "G2     19     45\n",
      "G3     31     70\n",
      "    False  True \n",
      "G1     54    107\n",
      "G2     23     41\n",
      "G3     38     63\n",
      "    histological_type,  vital_status,\n",
      "G1                   0              1\n",
      "G2                   1              0\n",
      "G3                   0              1\n",
      "    False  True \n",
      "G1    160      1\n",
      "G2     63      1\n",
      "G3    100      1\n",
      "    False  True \n",
      "G1    154      7\n",
      "G2     57      6\n",
      "G3     96      5\n",
      "    True   False\n",
      "G1    121     40\n",
      "G2     56      8\n",
      "G3     82     19\n",
      "    True   False\n",
      "G1    100     61\n",
      "G2     46     18\n",
      "G3     58     43\n",
      "    True   False\n",
      "G1    105     56\n",
      "G2     50     14\n",
      "G3     61     40\n",
      "    True   False\n",
      "G1    127     34\n",
      "G2     56      8\n",
      "G3     83     18\n",
      "    True   False\n",
      "G1    104     57\n",
      "G2     46     18\n",
      "G3     59     42\n",
      "    True   False\n",
      "G1    109     52\n",
      "G2     50     14\n",
      "G3     62     39\n",
      "    StandardRadiation  TMZChemoradiation.TMZChemoLong  NonstandardRadiation  \\\n",
      "G1                  8                              72                     5   \n",
      "G2                  7                              32                     2   \n",
      "G3                  3                              43                    11   \n",
      "\n",
      "    StandardRadiation.TMZChemoLong  StandardRadiation.TMZChemo  \\\n",
      "G1                              24                           0   \n",
      "G2                               8                           3   \n",
      "G3                              14                           3   \n",
      "\n",
      "    StandardRadiation.AlkChemoLong  TMZChemoLong  \\\n",
      "G1                               7             4   \n",
      "G2                               1             1   \n",
      "G3                               1             2   \n",
      "\n",
      "    NonstandardRadiation.AlkChemoLong  NonstandardRadiation.TMZChemoLong  \\\n",
      "G1                                  2                                  7   \n",
      "G2                                  0                                  0   \n",
      "G3                                  0                                  4   \n",
      "\n",
      "    NonstandardRadiation.TMZChemo  AlkChemoradiation.AlkChemoLong  AlkChemo  \\\n",
      "G1                              1                               0         3   \n",
      "G2                              1                               2         0   \n",
      "G3                              4                               0         1   \n",
      "\n",
      "    StandardRadiation.AlkChemo  AlkChemoLong  \n",
      "G1                           1             1  \n",
      "G2                           0             0  \n",
      "G3                           0             0  \n",
      "    False  True \n",
      "G1     89     72\n",
      "G2     32     32\n",
      "G3     58     43\n",
      "    DECEASED  LIVING\n",
      "G1       124      36\n",
      "G2        40      24\n",
      "G3        67      33\n",
      "    NON G-CIMP  G-CIMP\n",
      "G1         161       4\n",
      "G2          42      24\n",
      "G3         108       5\n",
      "    Classical  Proneural  Mesenchymal  Neural\n",
      "G1         68         25           38      34\n",
      "G2          4         48           12       5\n",
      "G3         23         18           55      17\n",
      "    True   False\n",
      "G1     21    144\n",
      "G2     13     56\n",
      "G3     24     89\n",
      "    TCGA-02-0001-01  TCGA-02-0003-01  TCGA-02-0007-01  TCGA-02-0009-01  \\\n",
      "G1                0                0                1                1   \n",
      "G2                0                0                0                0   \n",
      "G3                1                1                0                0   \n",
      "\n",
      "    TCGA-02-0010-01  TCGA-02-0011-01  TCGA-02-0014-01  TCGA-02-0021-01  \\\n",
      "G1                0                0                0                1   \n",
      "G2                1                1                1                0   \n",
      "G3                0                0                0                0   \n",
      "\n",
      "    TCGA-02-0024-01  TCGA-02-0027-01  ...  TCGA-76-4931-01  TCGA-76-4934-01  \\\n",
      "G1                0                0  ...                1                0   \n",
      "G2                1                0  ...                0                1   \n",
      "G3                0                1  ...                0                0   \n",
      "\n",
      "    TCGA-76-4935-01  TCGA-76-6191-01  TCGA-76-6192-01  TCGA-76-6193-01  \\\n",
      "G1                1                0                0                0   \n",
      "G2                0                0                0                0   \n",
      "G3                0                1                1                1   \n",
      "\n",
      "    TCGA-76-6282-01  TCGA-76-6285-01  TCGA-81-5910-01  TCGA-87-5896-01  \n",
      "G1                0                0                0                1  \n",
      "G2                0                0                0                0  \n",
      "G3                1                1                1                0  \n",
      "\n",
      "[3 rows x 347 columns]\n",
      "    High  GBM  Kirc+  Squamous  Iq  BRCA-LUAD+  Quiet  COAD-READ\n",
      "G1     0  143     11         0   1          10      0          0\n",
      "G2     4   25     22         8   1           8      0          1\n",
      "G3     2   80     18         0   0           7      5          1\n",
      "    C10-GBM  C1-LUAD-enriched  C5-KIRC\n",
      "G1       69                 0        0\n",
      "G2       26                 2        0\n",
      "G3       57                 1        2\n",
      "    cluster 1  cluster 3  cluster 4  cluster 5  cluster 6  cluster 2\n",
      "G1         12         66         36          3         13         31\n",
      "G2          3         12          8         21         19          6\n",
      "G3         35         15          6          4          6         46\n",
      "    GBM non-CIMP c15  GBM CIMP c16  Low purity c2  Cluster 5\n",
      "G1               158             3              0          0\n",
      "G2                46            21              0          2\n",
      "G3                97             4              9          1\n",
      "    KIRC-like c1  GBM-like c2  squamous c3\n",
      "G1             3           60            2\n",
      "G2             2           26            1\n",
      "G3             8           40            1\n",
      "    mutation cluster 8  mutation cluster 3  mutation cluster 4  \\\n",
      "G1                   5                   1                   3   \n",
      "G2                   1                   4                   2   \n",
      "G3                   7                   4                   8   \n",
      "\n",
      "    mutation c1uster 9  mutation cluster 5  mutation c1uster 10  \\\n",
      "G1                  27                  10                    1   \n",
      "G2                   5                  11                    1   \n",
      "G3                  27                   4                    2   \n",
      "\n",
      "    mutation cluster 14  mutation cluster 12  mutation cluster 7  \\\n",
      "G1                   27                    4                   2   \n",
      "G2                    2                    1                   1   \n",
      "G3                   19                    0                   2   \n",
      "\n",
      "    mutation cluster 13  mutation cluster 11  \n",
      "G1                    1                    1  \n",
      "G2                    2                    1  \n",
      "G3                    2                    1  \n",
      "    TCGA-02-0001  TCGA-02-0003  TCGA-02-0007  TCGA-02-0009  TCGA-02-0010  \\\n",
      "G1             0             0             1             1             0   \n",
      "G2             0             0             0             0             1   \n",
      "G3             1             1             0             0             0   \n",
      "\n",
      "    TCGA-02-0011  TCGA-02-0014  TCGA-02-0021  TCGA-02-0024  TCGA-02-0027  ...  \\\n",
      "G1             0             0             1             0             0  ...   \n",
      "G2             1             1             0             1             0  ...   \n",
      "G3             0             0             0             0             1  ...   \n",
      "\n",
      "    TCGA-76-4931  TCGA-76-4934  TCGA-76-4935  TCGA-76-6191  TCGA-76-6192  \\\n",
      "G1             1             0             1             0             0   \n",
      "G2             0             1             0             0             0   \n",
      "G3             0             0             0             1             1   \n",
      "\n",
      "    TCGA-76-6193  TCGA-76-6282  TCGA-76-6285  TCGA-81-5910  TCGA-87-5896  \n",
      "G1             0             0             0             0             1  \n",
      "G2             0             0             0             0             0  \n",
      "G3             1             1             1             1             0  \n",
      "\n",
      "[3 rows x 347 columns]\n",
      "    NO  YES\n",
      "G1  87   60\n",
      "G2  43   17\n",
      "G3  60   27\n",
      "     NO  YES\n",
      "G1  126   24\n",
      "G2   55    6\n",
      "G3   84    7\n",
      "     NO  YES\n",
      "G1  148    2\n",
      "G2   60    1\n",
      "G3   91    0\n",
      "    NO  YES\n",
      "G1  85   67\n",
      "G2  41   21\n",
      "G3  65   32\n",
      "     NO  YES\n",
      "G1  133   20\n",
      "G2   58    4\n",
      "G3   82   14\n",
      "    TCGA-02-0001-F22169  TCGA-02-0003-F22176  TCGA-02-0007-F22191  \\\n",
      "G1                    0                    0                    1   \n",
      "G2                    0                    0                    0   \n",
      "G3                    1                    1                    0   \n",
      "\n",
      "    TCGA-02-0009-F22196  TCGA-02-0010-F22198  TCGA-02-0011-F22204  \\\n",
      "G1                    1                    0                    0   \n",
      "G2                    0                    1                    1   \n",
      "G3                    0                    0                    0   \n",
      "\n",
      "    TCGA-02-0014-F22211  TCGA-02-0021-F22225  TCGA-02-0024-F22238  \\\n",
      "G1                    0                    1                    0   \n",
      "G2                    1                    0                    1   \n",
      "G3                    0                    0                    0   \n",
      "\n",
      "    TCGA-02-0027-F22254  ...  TCGA-76-4931-F28535  TCGA-76-4934-F28543  \\\n",
      "G1                    0  ...                    1                    0   \n",
      "G2                    0  ...                    0                    1   \n",
      "G3                    1  ...                    0                    0   \n",
      "\n",
      "    TCGA-76-4935-F28545  TCGA-76-6191-F28550  TCGA-76-6192-F28557  \\\n",
      "G1                    1                    0                    0   \n",
      "G2                    0                    0                    0   \n",
      "G3                    0                    1                    1   \n",
      "\n",
      "    TCGA-76-6193-F28561  TCGA-76-6282-F28565  TCGA-76-6285-F28570  \\\n",
      "G1                    0                    0                    0   \n",
      "G2                    0                    0                    0   \n",
      "G3                    1                    1                    1   \n",
      "\n",
      "    TCGA-81-5910-F28600  TCGA-87-5896-F43207  \n",
      "G1                    0                    1  \n",
      "G2                    0                    0  \n",
      "G3                    1                    0  \n",
      "\n",
      "[3 rows x 340 columns]\n",
      "    TCGA-02-0001  TCGA-02-0003  TCGA-02-0007  TCGA-02-0009  TCGA-02-0010  \\\n",
      "G1             0             0             1             1             0   \n",
      "G2             0             0             0             0             1   \n",
      "G3             1             1             0             0             0   \n",
      "\n",
      "    TCGA-02-0011  TCGA-02-0014  TCGA-02-0021  TCGA-02-0024  TCGA-02-0027  ...  \\\n",
      "G1             0             0             1             0             0  ...   \n",
      "G2             1             1             0             1             0  ...   \n",
      "G3             0             0             0             0             1  ...   \n",
      "\n",
      "    TCGA-76-4931  TCGA-76-4934  TCGA-76-4935  TCGA-76-6191  TCGA-76-6192  \\\n",
      "G1             1             0             1             0             0   \n",
      "G2             0             1             0             0             0   \n",
      "G3             0             0             0             1             1   \n",
      "\n",
      "    TCGA-76-6193  TCGA-76-6282  TCGA-76-6285  TCGA-81-5910  TCGA-87-5896  \n",
      "G1             0             0             0             0             1  \n",
      "G2             0             0             0             0             0  \n",
      "G3             1             1             1             1             0  \n",
      "\n",
      "[3 rows x 344 columns]\n",
      "    TCGA-02-0001-01C  TCGA-02-0003-01A  TCGA-02-0009-01A  TCGA-02-0010-01A  \\\n",
      "G1                 0                 0                 1                 0   \n",
      "G2                 0                 0                 0                 1   \n",
      "G3                 1                 1                 0                 0   \n",
      "\n",
      "    TCGA-02-0011-01B  TCGA-02-0021-01A  TCGA-02-0024-01B  TCGA-02-0027-01A  \\\n",
      "G1                 0                 1                 0                 0   \n",
      "G2                 1                 0                 1                 0   \n",
      "G3                 0                 0                 0                 1   \n",
      "\n",
      "    TCGA-02-0033-01A  TCGA-02-0034-01A  ...  TCGA-76-4929-01A  \\\n",
      "G1                 0                 0  ...                 0   \n",
      "G2                 0                 0  ...                 1   \n",
      "G3                 1                 1  ...                 0   \n",
      "\n",
      "    TCGA-76-4931-01A  TCGA-76-4934-01A  TCGA-76-6191-01A  TCGA-76-6192-01A  \\\n",
      "G1                 1                 0                 0                 0   \n",
      "G2                 0                 1                 0                 0   \n",
      "G3                 0                 0                 1                 1   \n",
      "\n",
      "    TCGA-76-6193-01A  TCGA-76-6282-01A  TCGA-76-6285-01A  TCGA-81-5910-01A  \\\n",
      "G1                 0                 0                 0                 0   \n",
      "G2                 0                 0                 0                 0   \n",
      "G3                 1                 1                 1                 1   \n",
      "\n",
      "    TCGA-87-5896-01A  \n",
      "G1                 1  \n",
      "G2                 0  \n",
      "G3                 0  \n",
      "\n",
      "[3 rows x 279 columns]\n",
      "    YES  NO\n",
      "G1  124  25\n",
      "G2   53   9\n",
      "G3   78  17\n",
      "    358  144  705  322  1077  630  2512  2362  1615  370  ...  290  135  146  \\\n",
      "G1    0    0    1    1     0    0     0     1     0    0  ...    1    1    0   \n",
      "G2    0    0    0    0     1    1     1     0     1    0  ...    0    0    1   \n",
      "G3    1    1    0    0     0    0     0     0     0    1  ...    0    0    0   \n",
      "\n",
      "    94  279  1121  508  82  519  49  \n",
      "G1   0    1     1    0   0    0   0  \n",
      "G2   0    0     0    0   0    0   0  \n",
      "G3   1    0     0    1   1    1   1  \n",
      "\n",
      "[3 rows x 237 columns]\n",
      "    137  40  518  264  351  145  2243  255  1401  257  ...  26  122  210  34  \\\n",
      "G1    0   0    2    1    1    0     0    1     0    0  ...   0    1    0   0   \n",
      "G2    0   0    0    0    1    1     1    0     1    0  ...   1    0    1   0   \n",
      "G3    1   1    0    0    1    2     0    0     0    1  ...   0    0    0   1   \n",
      "\n",
      "    112  [Discrepancy]  323  76  64  458  \n",
      "G1    1              1    0   0   0    0  \n",
      "G2    0              0    0   0   0    0  \n",
      "G3    0              0    1   1   1    1  \n",
      "\n",
      "[3 rows x 175 columns]\n",
      "    518  264  351  144  2243  254  257  1921  32  386  ...  265  233  488  \\\n",
      "G1    2    1    1    0     0    1    0     0   0    0  ...    0    0    0   \n",
      "G2    0    0    1    1     1    0    0     1   0    0  ...    0    0    0   \n",
      "G3    0    0    0    0     0    0    1     0   1    1  ...    1    1    1   \n",
      "\n",
      "    131  181  118  26  122  210  34  \n",
      "G1    2    1    0   0    1    0   0  \n",
      "G2    0    0    1   1    0    1   0  \n",
      "G3    0    0    0   0    0    0   1  \n",
      "\n",
      "[3 rows x 117 columns]\n",
      "    Scheduled Follow-up Submission  Additional New Tumor Event\n",
      "G1                              14                           1\n",
      "G2                               6                           0\n",
      "G3                               6                           0\n",
      "    Progressive Disease  Complete Remission/Response\n",
      "G1                    3                            1\n",
      "G2                    1                            0\n",
      "G3                    3                            0\n",
      "    2008-12-16  2008-12-17  2009-4-6  2010-5-6  2008-12-10  2008-12-8  \\\n",
      "G1           9           2         0         2           6          1   \n",
      "G2           7           4         0         2           1          1   \n",
      "G3          10           6         1         1           6          1   \n",
      "\n",
      "    2013-11-26  2013-2-12  2008-12-11  2009-5-11  ...  2012-4-3  2010-6-3  \\\n",
      "G1           1          1           2          3  ...         1         2   \n",
      "G2           0          0           0          4  ...         1         1   \n",
      "G3           0          1           1          0  ...         0         1   \n",
      "\n",
      "    2010-10-26  2011-5-5  2011-7-21  2011-4-6  2015-3-26  2011-9-22  \\\n",
      "G1           2         0          0         1          1          0   \n",
      "G2           0         0          1         3          0          0   \n",
      "G3           1         1          0         2          0          4   \n",
      "\n",
      "    2011-10-21  2011-6-7  \n",
      "G1           0         0  \n",
      "G2           0         0  \n",
      "G3           1         1  \n",
      "\n",
      "[3 rows x 99 columns]\n",
      "    FEMALE  MALE\n",
      "G1      68    96\n",
      "G2      35    34\n",
      "G3      40    71\n",
      "    Untreated primary (de novo) GBM  Treated primary GBM  \\\n",
      "G1                              156                    7   \n",
      "G2                               63                    6   \n",
      "G3                              104                    5   \n",
      "\n",
      "    Glioblastoma Multiforme (GBM)  \n",
      "G1                              1  \n",
      "G2                              0  \n",
      "G3                              2  \n",
      "    Yes   No\n",
      "G1    8  156\n",
      "G2    5   64\n",
      "G3    6  105\n",
      "     NO  YES\n",
      "G1  115   38\n",
      "G2   48   14\n",
      "G3   87    8\n",
      "    C71.9  C71.2\n",
      "G1    164      0\n",
      "G2     69      0\n",
      "G3    110      1\n",
      "    C71.9  C71.2\n",
      "G1    164      0\n",
      "G2     69      0\n",
      "G3    110      1\n",
      "     NO  YES\n",
      "G1  147    6\n",
      "G2   60    2\n",
      "G3   92    3\n",
      "    Tumor resection  Excisional Biopsy  Incisional Biopsy  \\\n",
      "G1              131                 28                  2   \n",
      "G2               64                  4                  1   \n",
      "G3              104                  5                  1   \n",
      "\n",
      "    Fine needle aspiration biopsy  Other method, specify:  \n",
      "G1                              2                       1  \n",
      "G2                              0                       0  \n",
      "G3                              1                       0  \n",
      "    YES  NO\n",
      "G1    1   9\n",
      "G2    0   3\n",
      "G3    0   4\n",
      "    Recurrence  Progression of Disease  Locoregional Disease\n",
      "G1          37                      70                     1\n",
      "G2          11                      28                     2\n",
      "G3          19                      45                     2\n",
      "    NO  YES\n",
      "G1   3    0\n",
      "G2   0    1\n",
      "G3   1    2\n",
      "    TCGA-02-0001.392585DD-C29D-4226-B34D-1CAB66C10009.pdf  \\\n",
      "G1                                                  0       \n",
      "G2                                                  0       \n",
      "G3                                                  1       \n",
      "\n",
      "    TCGA-02-0003.1E5D6194-FE78-41E6-A05E-92B2A8E0B162.pdf  \\\n",
      "G1                                                  0       \n",
      "G2                                                  0       \n",
      "G3                                                  1       \n",
      "\n",
      "    TCGA-02-0009.1AD9AD8E-0DA4-4D15-B984-A59D88014D51.pdf  \\\n",
      "G1                                                  1       \n",
      "G2                                                  0       \n",
      "G3                                                  0       \n",
      "\n",
      "    TCGA-02-0010.C7B6B2FC-48A7-4A56-893F-F8EDAFCE952A.pdf  \\\n",
      "G1                                                  0       \n",
      "G2                                                  1       \n",
      "G3                                                  0       \n",
      "\n",
      "    TCGA-02-0011.49241786-AEE6-4404-B696-D1AB8BAEE6F4.pdf  \\\n",
      "G1                                                  0       \n",
      "G2                                                  1       \n",
      "G3                                                  0       \n",
      "\n",
      "    TCGA-02-0021.1BEE82DA-EB4A-492E-B872-632F81F7C84E.pdf  \\\n",
      "G1                                                  1       \n",
      "G2                                                  0       \n",
      "G3                                                  0       \n",
      "\n",
      "    TCGA-02-0024.4F8AA412-3AAE-47F2-9F07-F4A9B2EE11DF.pdf  \\\n",
      "G1                                                  0       \n",
      "G2                                                  1       \n",
      "G3                                                  0       \n",
      "\n",
      "    TCGA-02-0027.2E9F38EF-D6A4-4431-8514-08FEFF62AA68.pdf  \\\n",
      "G1                                                  0       \n",
      "G2                                                  0       \n",
      "G3                                                  1       \n",
      "\n",
      "    TCGA-02-0033.BD51E951-1FAF-4FCA-8985-CF63B76411F6.pdf  \\\n",
      "G1                                                  0       \n",
      "G2                                                  0       \n",
      "G3                                                  1       \n",
      "\n",
      "    TCGA-02-0034.834B13FD-1F37-4B25-AD92-90A62D6495B8.pdf  ...  \\\n",
      "G1                                                  0      ...   \n",
      "G2                                                  0      ...   \n",
      "G3                                                  1      ...   \n",
      "\n",
      "    TCGA-76-4929.65054ac8-f9aa-4342-9f18-81c15efd6d81.pdf  \\\n",
      "G1                                                  0       \n",
      "G2                                                  1       \n",
      "G3                                                  0       \n",
      "\n",
      "    TCGA-76-4931.f34d64d6-6103-407f-b45b-4e660a1bfcd1.pdf  \\\n",
      "G1                                                  1       \n",
      "G2                                                  0       \n",
      "G3                                                  0       \n",
      "\n",
      "    TCGA-76-4934.31727bd3-2af9-4648-b5ae-60af7be49ea7.pdf  \\\n",
      "G1                                                  0       \n",
      "G2                                                  1       \n",
      "G3                                                  0       \n",
      "\n",
      "    TCGA-76-6191.d91aa8ca-fe4b-4f2e-89a7-c56d6fe176e1.pdf  \\\n",
      "G1                                                  0       \n",
      "G2                                                  0       \n",
      "G3                                                  1       \n",
      "\n",
      "    TCGA-76-6192.d2a94594-7f5a-40f2-87fa-beee3fa5be5b.pdf  \\\n",
      "G1                                                  0       \n",
      "G2                                                  0       \n",
      "G3                                                  1       \n",
      "\n",
      "    TCGA-76-6193.1ee55ecb-8de5-4a21-9fa0-babcb2d07212.pdf  \\\n",
      "G1                                                  0       \n",
      "G2                                                  0       \n",
      "G3                                                  1       \n",
      "\n",
      "    TCGA-76-6282.fe3b0aee-6477-48dd-af17-54f991fa47a5.pdf  \\\n",
      "G1                                                  0       \n",
      "G2                                                  0       \n",
      "G3                                                  1       \n",
      "\n",
      "    TCGA-76-6285.9b0a966e-0cd8-44a2-8f15-c5f5ed40c189.pdf  \\\n",
      "G1                                                  0       \n",
      "G2                                                  0       \n",
      "G3                                                  1       \n",
      "\n",
      "    TCGA-81-5910.9d6e4a44-727e-4713-842e-a75ded996614.pdf  \\\n",
      "G1                                                  0       \n",
      "G2                                                  0       \n",
      "G3                                                  1       \n",
      "\n",
      "    TCGA-87-5896.a743b83f-a4ff-4b06-9728-765a6bb6807f.pdf  \n",
      "G1                                                  1      \n",
      "G2                                                  0      \n",
      "G3                                                  0      \n",
      "\n",
      "[3 rows x 279 columns]\n",
      "    Post-Adjuvant Therapy  Pre-Adjuvant Therapy  Preoperative  Other\n",
      "G1                     29                    51            33      4\n",
      "G2                     10                    18            10      6\n",
      "G3                     20                    24            16      2\n",
      "    WITH TUMOR  TUMOR FREE\n",
      "G1         138           8\n",
      "G2          50           6\n",
      "G3          96           5\n",
      "    YES  NO\n",
      "G1  134  16\n",
      "G2   59   5\n",
      "G3   89  16\n",
      "    YES   NO  [Discrepancy]\n",
      "G1    8  152              1\n",
      "G2    5   59              0\n",
      "G3    6   95              0\n",
      "    Complete Remission/Response  Progressive Disease  Stable Disease\n",
      "G1                            1                    3               0\n",
      "G2                            0                    0               1\n",
      "G3                            1                    2               2\n",
      "     NO  YES\n",
      "G1  161    3\n",
      "G2   66    3\n",
      "G3  107    4\n",
      "    YES  NO\n",
      "G1  133  23\n",
      "G2   60   7\n",
      "G3   93  14\n",
      "     NO  YES\n",
      "G1  126   25\n",
      "G2   56    6\n",
      "G3   85    8\n",
      "    02  06  12  14  15  16  19  26  27  28  32  41  76  81  87\n",
      "G1  13  29  31  20   1   8  18   7  14   6  10   4   2   0   1\n",
      "G2  13  11   3  12   2   3   5   2   2   7   4   2   3   0   0\n",
      "G3  18  29   3   6   1   2  12   3   1  14  11   3   7   1   0\n",
      "    C    A   B\n",
      "G1  1  115  12\n",
      "G2  2   51   4\n",
      "G3  2   87   5\n",
      "    DECEASED  LIVING\n",
      "G1       138      25\n",
      "G2        48      21\n",
      "G3        84      27\n",
      "      3    1   2\n",
      "G1    0  165   0\n",
      "G2    0    0  69\n",
      "G3  113    0   0\n",
      "    353.0   144.0   705.0   322.0   1077.0  630.0   2511.0  2361.0  1614.0  \\\n",
      "G1       0       0       1       1       0       0       0       1       0   \n",
      "G2       0       0       0       0       1       1       1       0       1   \n",
      "G3       1       1       0       0       0       0       0       0       0   \n",
      "\n",
      "    370.0   ...  330.0   26.0    406.0   272.0   290.0   135.0   351.0   \\\n",
      "G1       0  ...       0       0       1       0       1       1       0   \n",
      "G2       0  ...       0       1       0       0       0       0       1   \n",
      "G3       1  ...       1       0       0       1       0       0       0   \n",
      "\n",
      "    94.0    49.0    2.0     \n",
      "G1       0       0       1  \n",
      "G2       0       0       0  \n",
      "G3       1       1       0  \n",
      "\n",
      "[3 rows x 276 columns]\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "[('CDE_therapy', 0.0239712693970924),\n",
       " ('CDE_vital_status', 0.04219955419397384),\n",
       " ('G_CIMP_STATUS', 1.897978420038419e-15),\n",
       " ('GeneExp_Subtype', 8.534673419184012e-22),\n",
       " ('_PANCAN_CNA_PANCAN_K8', 2.1989441476890682e-14),\n",
       " ('_PANCAN_DNAMethyl_GBM', 9.889506786244145e-30),\n",
       " ('_PANCAN_DNAMethyl_PANCAN', 2.253317472999156e-15),\n",
       " ('_PANCAN_mutation_PANCAN', 0.002746407829379185),\n",
       " ('form_completion_date', 0.04856771110301229),\n",
       " ('hormonal_therapy', 0.004905154740707577),\n",
       " ('initial_pathologic_diagnosis_method', 0.04956822543700062),\n",
       " ('tissue_source_site', 7.59186908651961e-05),\n",
       " ('vital_status', 0.023497201679813208),\n",
       " ('subtype', 6.940552177863327e-149)]"
      ]
     },
     "execution_count": 213,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sig_d =[]\n",
    "for key,value in name_count.items():\n",
    "    if value.size ==0 or value.shape[1]<2:\n",
    "        continue\n",
    "    print(name_count[key])\n",
    "    chi2,pval,dof,exp  = chi2_contingency(name_count[key])\n",
    "   \n",
    "    if pval<=0.05:\n",
    "        sig_d.append((key, pval))\n",
    "sig_d\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>StandardRadiation</th>\n",
       "      <th>TMZChemoradiation.TMZChemoLong</th>\n",
       "      <th>NonstandardRadiation</th>\n",
       "      <th>StandardRadiation.TMZChemoLong</th>\n",
       "      <th>StandardRadiation.TMZChemo</th>\n",
       "      <th>StandardRadiation.AlkChemoLong</th>\n",
       "      <th>TMZChemoLong</th>\n",
       "      <th>NonstandardRadiation.AlkChemoLong</th>\n",
       "      <th>NonstandardRadiation.TMZChemoLong</th>\n",
       "      <th>NonstandardRadiation.TMZChemo</th>\n",
       "      <th>AlkChemoradiation.AlkChemoLong</th>\n",
       "      <th>AlkChemo</th>\n",
       "      <th>StandardRadiation.AlkChemo</th>\n",
       "      <th>AlkChemoLong</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>G1</td>\n",
       "      <td>8</td>\n",
       "      <td>72</td>\n",
       "      <td>5</td>\n",
       "      <td>24</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>G2</td>\n",
       "      <td>7</td>\n",
       "      <td>32</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>G3</td>\n",
       "      <td>3</td>\n",
       "      <td>43</td>\n",
       "      <td>11</td>\n",
       "      <td>14</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    StandardRadiation  TMZChemoradiation.TMZChemoLong  NonstandardRadiation  \\\n",
       "G1                  8                              72                     5   \n",
       "G2                  7                              32                     2   \n",
       "G3                  3                              43                    11   \n",
       "\n",
       "    StandardRadiation.TMZChemoLong  StandardRadiation.TMZChemo  \\\n",
       "G1                              24                           0   \n",
       "G2                               8                           3   \n",
       "G3                              14                           3   \n",
       "\n",
       "    StandardRadiation.AlkChemoLong  TMZChemoLong  \\\n",
       "G1                               7             4   \n",
       "G2                               1             1   \n",
       "G3                               1             2   \n",
       "\n",
       "    NonstandardRadiation.AlkChemoLong  NonstandardRadiation.TMZChemoLong  \\\n",
       "G1                                  2                                  7   \n",
       "G2                                  0                                  0   \n",
       "G3                                  0                                  4   \n",
       "\n",
       "    NonstandardRadiation.TMZChemo  AlkChemoradiation.AlkChemoLong  AlkChemo  \\\n",
       "G1                              1                               0         3   \n",
       "G2                              1                               2         0   \n",
       "G3                              4                               0         1   \n",
       "\n",
       "    StandardRadiation.AlkChemo  AlkChemoLong  \n",
       "G1                           1             1  \n",
       "G2                           0             0  \n",
       "G3                           0             0  "
      ]
     },
     "execution_count": 215,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_count['CDE_therapy']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DECEASED</th>\n",
       "      <th>LIVING</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>G1</td>\n",
       "      <td>124</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>G2</td>\n",
       "      <td>40</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>G3</td>\n",
       "      <td>67</td>\n",
       "      <td>33</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    DECEASED  LIVING\n",
       "G1       124      36\n",
       "G2        40      24\n",
       "G3        67      33"
      ]
     },
     "execution_count": 216,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_count['CDE_vital_status']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 217,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NON G-CIMP</th>\n",
       "      <th>G-CIMP</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>G1</td>\n",
       "      <td>161</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>G2</td>\n",
       "      <td>42</td>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>G3</td>\n",
       "      <td>108</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    NON G-CIMP  G-CIMP\n",
       "G1         161       4\n",
       "G2          42      24\n",
       "G3         108       5"
      ]
     },
     "execution_count": 217,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_count['G_CIMP_STATUS']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Recently, a subset of relatively favorable prognosis GBMs has been identified. These glioma CpG island methylator phenotype, or G-CIMP tumors, have distinct genomic copy number aberrations, DNA methylation patterns, and (mRNA) expression profiles compared to other GBMs. While the standard method for identifying G-CIMP tumors is based on genome-wide DNA methylation data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Classical</th>\n",
       "      <th>Proneural</th>\n",
       "      <th>Mesenchymal</th>\n",
       "      <th>Neural</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>G1</td>\n",
       "      <td>68</td>\n",
       "      <td>25</td>\n",
       "      <td>38</td>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>G2</td>\n",
       "      <td>4</td>\n",
       "      <td>48</td>\n",
       "      <td>12</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>G3</td>\n",
       "      <td>23</td>\n",
       "      <td>18</td>\n",
       "      <td>55</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Classical  Proneural  Mesenchymal  Neural\n",
       "G1         68         25           38      34\n",
       "G2          4         48           12       5\n",
       "G3         23         18           55      17"
      ]
     },
     "execution_count": 218,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_count['GeneExp_Subtype']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>High</th>\n",
       "      <th>GBM</th>\n",
       "      <th>Kirc+</th>\n",
       "      <th>Squamous</th>\n",
       "      <th>Iq</th>\n",
       "      <th>BRCA-LUAD+</th>\n",
       "      <th>Quiet</th>\n",
       "      <th>COAD-READ</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>G1</td>\n",
       "      <td>0</td>\n",
       "      <td>143</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>G2</td>\n",
       "      <td>4</td>\n",
       "      <td>25</td>\n",
       "      <td>22</td>\n",
       "      <td>8</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>G3</td>\n",
       "      <td>2</td>\n",
       "      <td>80</td>\n",
       "      <td>18</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>7</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    High  GBM  Kirc+  Squamous  Iq  BRCA-LUAD+  Quiet  COAD-READ\n",
       "G1     0  143     11         0   1          10      0          0\n",
       "G2     4   25     22         8   1           8      0          1\n",
       "G3     2   80     18         0   0           7      5          1"
      ]
     },
     "execution_count": 219,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_count['_PANCAN_CNA_PANCAN_K8']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>cluster 1</th>\n",
       "      <th>cluster 3</th>\n",
       "      <th>cluster 4</th>\n",
       "      <th>cluster 5</th>\n",
       "      <th>cluster 6</th>\n",
       "      <th>cluster 2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>G1</td>\n",
       "      <td>12</td>\n",
       "      <td>66</td>\n",
       "      <td>36</td>\n",
       "      <td>3</td>\n",
       "      <td>13</td>\n",
       "      <td>31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>G2</td>\n",
       "      <td>3</td>\n",
       "      <td>12</td>\n",
       "      <td>8</td>\n",
       "      <td>21</td>\n",
       "      <td>19</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>G3</td>\n",
       "      <td>35</td>\n",
       "      <td>15</td>\n",
       "      <td>6</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>46</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    cluster 1  cluster 3  cluster 4  cluster 5  cluster 6  cluster 2\n",
       "G1         12         66         36          3         13         31\n",
       "G2          3         12          8         21         19          6\n",
       "G3         35         15          6          4          6         46"
      ]
     },
     "execution_count": 220,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_count['_PANCAN_DNAMethyl_GBM']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>NO</th>\n",
       "      <th>YES</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>G1</td>\n",
       "      <td>115</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>G2</td>\n",
       "      <td>48</td>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>G3</td>\n",
       "      <td>87</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     NO  YES\n",
       "G1  115   38\n",
       "G2   48   14\n",
       "G3   87    8"
      ]
     },
     "execution_count": 221,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_count['hormonal_therapy']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Tumor resection</th>\n",
       "      <th>Excisional Biopsy</th>\n",
       "      <th>Incisional Biopsy</th>\n",
       "      <th>Fine needle aspiration biopsy</th>\n",
       "      <th>Other method, specify:</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>G1</td>\n",
       "      <td>131</td>\n",
       "      <td>28</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>G2</td>\n",
       "      <td>64</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>G3</td>\n",
       "      <td>104</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Tumor resection  Excisional Biopsy  Incisional Biopsy  \\\n",
       "G1              131                 28                  2   \n",
       "G2               64                  4                  1   \n",
       "G3              104                  5                  1   \n",
       "\n",
       "    Fine needle aspiration biopsy  Other method, specify:  \n",
       "G1                              2                       1  \n",
       "G2                              0                       0  \n",
       "G3                              1                       0  "
      ]
     },
     "execution_count": 222,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_count['initial_pathologic_diagnosis_method']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 223,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>02</th>\n",
       "      <th>06</th>\n",
       "      <th>12</th>\n",
       "      <th>14</th>\n",
       "      <th>15</th>\n",
       "      <th>16</th>\n",
       "      <th>19</th>\n",
       "      <th>26</th>\n",
       "      <th>27</th>\n",
       "      <th>28</th>\n",
       "      <th>32</th>\n",
       "      <th>41</th>\n",
       "      <th>76</th>\n",
       "      <th>81</th>\n",
       "      <th>87</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>G1</td>\n",
       "      <td>13</td>\n",
       "      <td>29</td>\n",
       "      <td>31</td>\n",
       "      <td>20</td>\n",
       "      <td>1</td>\n",
       "      <td>8</td>\n",
       "      <td>18</td>\n",
       "      <td>7</td>\n",
       "      <td>14</td>\n",
       "      <td>6</td>\n",
       "      <td>10</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>G2</td>\n",
       "      <td>13</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>G3</td>\n",
       "      <td>18</td>\n",
       "      <td>29</td>\n",
       "      <td>3</td>\n",
       "      <td>6</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>12</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>14</td>\n",
       "      <td>11</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    02  06  12  14  15  16  19  26  27  28  32  41  76  81  87\n",
       "G1  13  29  31  20   1   8  18   7  14   6  10   4   2   0   1\n",
       "G2  13  11   3  12   2   3   5   2   2   7   4   2   3   0   0\n",
       "G3  18  29   3   6   1   2  12   3   1  14  11   3   7   1   0"
      ]
     },
     "execution_count": 223,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_count['tissue_source_site']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DECEASED</th>\n",
       "      <th>LIVING</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>G1</td>\n",
       "      <td>138</td>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>G2</td>\n",
       "      <td>48</td>\n",
       "      <td>21</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>G3</td>\n",
       "      <td>84</td>\n",
       "      <td>27</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    DECEASED  LIVING\n",
       "G1       138      25\n",
       "G2        48      21\n",
       "G3        84      27"
      ]
     },
     "execution_count": 224,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "name_count['vital_status']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "metadata": {},
   "outputs": [],
   "source": [
    "cnv = pd.read_csv('data/cnv.txt',delimiter= '\\t',index_col=0,header=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>x</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>343</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>344</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>345</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>346</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>347</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>347 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     x\n",
       "1    3\n",
       "2    3\n",
       "3    1\n",
       "4    1\n",
       "5    2\n",
       "..  ..\n",
       "343  3\n",
       "344  3\n",
       "345  3\n",
       "346  3\n",
       "347  1\n",
       "\n",
       "[347 rows x 1 columns]"
      ]
     },
     "execution_count": 231,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 240,
   "metadata": {},
   "outputs": [],
   "source": [
    "cnv = cnv.merge(group,how='inner',left_index=True, right_index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 241,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['ACAP3', 'ACTRT2', 'AGRN', 'ANKRD65', 'ATAD3A', 'ATAD3B', 'ATAD3C',\n",
       "       'AURKAIP1', 'B3GALT6', 'C1orf159',\n",
       "       ...\n",
       "       'SNORA56', 'TMLHE', 'VBP1', 'IL9R|ENSG00000124334.12',\n",
       "       'SPRY3|ENSG00000168939.6', 'VAMP7|ENSG00000124333.10',\n",
       "       'WASH6P|ENSG00000182484.10', 'WASIR1|ENSG00000185203.7', 'x_x', 'x_y'],\n",
       "      dtype='object', length=24778)"
      ]
     },
     "execution_count": 241,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cnv.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 244,
   "metadata": {},
   "outputs": [],
   "source": [
    "idh1 = cnv[['x_x', 'IDH1']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 248,
   "metadata": {},
   "outputs": [],
   "source": [
    "g2 = idh1[idh1['x_x']==2]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 250,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(69, 2)"
      ]
     },
     "execution_count": 250,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g2.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 252,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(63, 2)"
      ]
     },
     "execution_count": 252,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g2[g2['IDH1']!=0].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 251,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(165, 2)"
      ]
     },
     "execution_count": 251,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g1 = idh1[idh1['x_x']==1]\n",
    "g1.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 253,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(149, 2)"
      ]
     },
     "execution_count": 253,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g1[g1['IDH1']!=0].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 254,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(113, 2)"
      ]
     },
     "execution_count": 254,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g3 = idh1[idh1['x_x']==3]\n",
    "g3.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 255,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(101, 2)"
      ]
     },
     "execution_count": 255,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "g3[g3['IDH1']!=0].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 256,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[0.9130434782608695, 0.9030303030303031, 0.8938053097345132]"
      ]
     },
     "execution_count": 256,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[63/69,149/165,101/113]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
